// @dynamic-labs/embedded-wallet/src/index.cjs
'use client'
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var TurnkeyWalletConnectorBase = require('./lib/TurnkeyWalletConnectorBase/TurnkeyWalletConnectorBase.cjs');
var types = require('./types.cjs');
var PasskeyService = require('./lib/utils/PasskeyService/PasskeyService.cjs');
var findTurnkeyVerifiedCredentials = require('./lib/utils/findTurnkeyVerifiedCredentials/findTurnkeyVerifiedCredentials.cjs');
require('./lib/utils/convertAttestationTransports/convertAttestationTransports.cjs');
var logger = require('./lib/utils/logger/logger.cjs');
var TurnkeyAuthenticatorRecoveryHandler = require('./lib/AuthenticatorHandler/TurnkeyAuthenticatorRecoveryHandler.cjs');
var ExportHandler = require('./lib/ExportHandler/ExportHandler.cjs');
var constants = require('./lib/constants.cjs');



exports.TurnkeyWalletConnectorBase = TurnkeyWalletConnectorBase.TurnkeyWalletConnectorBase;
exports.TurnkeyWalletConnectorInfo = types.TurnkeyWalletConnectorInfo;
exports.PasskeyService = PasskeyService.PasskeyService;
exports.findTurnkeyVerifiedCredentials = findTurnkeyVerifiedCredentials.findTurnkeyVerifiedCredentials;
exports.findTurnkeyVerifiedCredentialsWithSmartWalletRef = findTurnkeyVerifiedCredentials.findTurnkeyVerifiedCredentialsWithSmartWalletRef;
exports.logger = logger.logger;
exports.turnkeyAuthenticatorRecoveryHandler = TurnkeyAuthenticatorRecoveryHandler.turnkeyAuthenticatorRecoveryHandler;
exports.ExportHandler = ExportHandler.ExportHandler;
exports.INVALID_PASSKEY_SELECTED = constants.INVALID_PASSKEY_SELECTED;
exports.TURNKEY_API_BASE_URL = constants.TURNKEY_API_BASE_URL;
exports.TURNKEY_API_KEY_EXPIRY_MESSAGE = constants.TURNKEY_API_KEY_EXPIRY_MESSAGE;
exports.TURNKEY_API_KEY_NOT_FOUND_MESSAGE = constants.TURNKEY_API_KEY_NOT_FOUND_MESSAGE;
exports.TURNKEY_SDK_BENIGN_ERRORS = constants.TURNKEY_SDK_BENIGN_ERRORS;
exports.TURNKEY_SDK_SESSION_KEY_RETRYABLE_ERRORS = constants.TURNKEY_SDK_SESSION_KEY_RETRYABLE_ERRORS;
exports.WEBAUTHN_NOT_SUPPORTED_OR_CANCELLED_ERROR_MESSAGE = constants.WEBAUTHN_NOT_SUPPORTED_OR_CANCELLED_ERROR_MESSAGE;
exports.WEBAUTHN_NOT_SUPPORTED_OR_DISABLED_ERROR_MESSAGE = constants.WEBAUTHN_NOT_SUPPORTED_OR_DISABLED_ERROR_MESSAGE;
