// @dynamic-labs/sdk-react-core/src/lib/shared/utils/functions/wallet/createWallet.cjs
'use client'
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var walletBook = require('@dynamic-labs/wallet-book');

const createWallet = (walletBook$1, wallet) => {
    var _a;
    const walletBookWallet = walletBook.findWalletBookWallet(walletBook$1, wallet.key);
    return {
        chainGroup: walletBookWallet === null || walletBookWallet === void 0 ? void 0 : walletBookWallet.chainGroup,
        group: (walletBookWallet === null || walletBookWallet === void 0 ? void 0 : walletBookWallet.chainGroup) || (walletBookWallet === null || walletBookWallet === void 0 ? void 0 : walletBookWallet.walletGroup),
        injectedConfig: walletBookWallet === null || walletBookWallet === void 0 ? void 0 : walletBookWallet.injectedConfig,
        isInstalledOnBrowser: (_a = wallet.isInstalledOnBrowser()) !== null && _a !== void 0 ? _a : false,
        key: wallet.key,
        name: wallet.name,
        walletConnector: wallet,
        walletGroup: walletBookWallet === null || walletBookWallet === void 0 ? void 0 : walletBookWallet.walletGroup,
    };
};

exports.createWallet = createWallet;
