// @hpke/core/script/mod.js
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "./src/aeads/aesGcm.js", "./src/aeads/exportOnly.js", "./src/errors.js", "./src/identifiers.js", "./src/native.js", "./src/kems/dhkem.js", "./src/kems/dhkemPrimitives/ec.js", "./src/kems/hybridkem.js", "./src/xCryptoKey.js", "./src/consts.js"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.INPUT_LENGTH_LIMIT = exports.XCryptoKey = exports.Hybridkem = exports.Ec = exports.Dhkem = exports.HkdfSha512 = exports.HkdfSha384 = exports.HkdfSha256 = exports.DhkemP521HkdfSha512 = exports.DhkemP384HkdfSha384 = exports.DhkemP256HkdfSha256 = exports.CipherSuite = exports.KemId = exports.KdfId = exports.AeadId = exports.ExportOnly = exports.Aes256Gcm = exports.Aes128Gcm = void 0;
    var aesGcm_js_1 = require("./src/aeads/aesGcm.js");
    Object.defineProperty(exports, "Aes128Gcm", { enumerable: true, get: function () { return aesGcm_js_1.Aes128Gcm; } });
    Object.defineProperty(exports, "Aes256Gcm", { enumerable: true, get: function () { return aesGcm_js_1.Aes256Gcm; } });
    var exportOnly_js_1 = require("./src/aeads/exportOnly.js");
    Object.defineProperty(exports, "ExportOnly", { enumerable: true, get: function () { return exportOnly_js_1.ExportOnly; } });
    __exportStar(require("./src/errors.js"), exports);
    var identifiers_js_1 = require("./src/identifiers.js");
    Object.defineProperty(exports, "AeadId", { enumerable: true, get: function () { return identifiers_js_1.AeadId; } });
    Object.defineProperty(exports, "KdfId", { enumerable: true, get: function () { return identifiers_js_1.KdfId; } });
    Object.defineProperty(exports, "KemId", { enumerable: true, get: function () { return identifiers_js_1.KemId; } });
    var native_js_1 = require("./src/native.js");
    Object.defineProperty(exports, "CipherSuite", { enumerable: true, get: function () { return native_js_1.CipherSuite; } });
    Object.defineProperty(exports, "DhkemP256HkdfSha256", { enumerable: true, get: function () { return native_js_1.DhkemP256HkdfSha256; } });
    Object.defineProperty(exports, "DhkemP384HkdfSha384", { enumerable: true, get: function () { return native_js_1.DhkemP384HkdfSha384; } });
    Object.defineProperty(exports, "DhkemP521HkdfSha512", { enumerable: true, get: function () { return native_js_1.DhkemP521HkdfSha512; } });
    Object.defineProperty(exports, "HkdfSha256", { enumerable: true, get: function () { return native_js_1.HkdfSha256; } });
    Object.defineProperty(exports, "HkdfSha384", { enumerable: true, get: function () { return native_js_1.HkdfSha384; } });
    Object.defineProperty(exports, "HkdfSha512", { enumerable: true, get: function () { return native_js_1.HkdfSha512; } });
    var dhkem_js_1 = require("./src/kems/dhkem.js");
    Object.defineProperty(exports, "Dhkem", { enumerable: true, get: function () { return dhkem_js_1.Dhkem; } });
    var ec_js_1 = require("./src/kems/dhkemPrimitives/ec.js");
    Object.defineProperty(exports, "Ec", { enumerable: true, get: function () { return ec_js_1.Ec; } });
    var hybridkem_js_1 = require("./src/kems/hybridkem.js");
    Object.defineProperty(exports, "Hybridkem", { enumerable: true, get: function () { return hybridkem_js_1.Hybridkem; } });
    var xCryptoKey_js_1 = require("./src/xCryptoKey.js");
    Object.defineProperty(exports, "XCryptoKey", { enumerable: true, get: function () { return xCryptoKey_js_1.XCryptoKey; } });
    var consts_js_1 = require("./src/consts.js");
    Object.defineProperty(exports, "INPUT_LENGTH_LIMIT", { enumerable: true, get: function () { return consts_js_1.INPUT_LENGTH_LIMIT; } });
});
