// @dynamic-labs/embedded-wallet/src/lib/constants.cjs
'use client'
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const TURNKEY_API_BASE_URL = 'https://api.turnkey.com';
const TURNKEY_API_KEY_EXPIRY_MESSAGE = 'Turnkey error 16: expired api key';
const TURNKEY_API_KEY_NOT_FOUND_MESSAGE = 'Turnkey error 16: could not find public key';
const WEBAUTHN_NOT_SUPPORTED_OR_CANCELLED_ERROR_MESSAGE = 'The operation either timed out or was not allowed';
const WEBAUTHN_NOT_SUPPORTED_OR_DISABLED_ERROR_MESSAGE = 'The request is not allowed by the user agent or the platform in the current context, possibly because the user denied permission.';
const INVALID_PASSKEY_SELECTED = 'Turnkey error 5: webauthn authenticator not found in organization or parent organization';
const TURNKEY_SDK_SESSION_KEY_RETRYABLE_ERRORS = [
    TURNKEY_API_KEY_EXPIRY_MESSAGE,
    TURNKEY_API_KEY_NOT_FOUND_MESSAGE,
];
const TURNKEY_SDK_BENIGN_ERRORS = [
    WEBAUTHN_NOT_SUPPORTED_OR_CANCELLED_ERROR_MESSAGE,
    WEBAUTHN_NOT_SUPPORTED_OR_DISABLED_ERROR_MESSAGE,
    INVALID_PASSKEY_SELECTED,
];

exports.INVALID_PASSKEY_SELECTED = INVALID_PASSKEY_SELECTED;
exports.TURNKEY_API_BASE_URL = TURNKEY_API_BASE_URL;
exports.TURNKEY_API_KEY_EXPIRY_MESSAGE = TURNKEY_API_KEY_EXPIRY_MESSAGE;
exports.TURNKEY_API_KEY_NOT_FOUND_MESSAGE = TURNKEY_API_KEY_NOT_FOUND_MESSAGE;
exports.TURNKEY_SDK_BENIGN_ERRORS = TURNKEY_SDK_BENIGN_ERRORS;
exports.TURNKEY_SDK_SESSION_KEY_RETRYABLE_ERRORS = TURNKEY_SDK_SESSION_KEY_RETRYABLE_ERRORS;
exports.WEBAUTHN_NOT_SUPPORTED_OR_CANCELLED_ERROR_MESSAGE = WEBAUTHN_NOT_SUPPORTED_OR_CANCELLED_ERROR_MESSAGE;
exports.WEBAUTHN_NOT_SUPPORTED_OR_DISABLED_ERROR_MESSAGE = WEBAUTHN_NOT_SUPPORTED_OR_DISABLED_ERROR_MESSAGE;
