// @dynamic-labs/sdk-react-core/src/lib/views/Passkey/EmbeddedAuthChoiceCards/EmbeddedAuthChoiceCards.js
'use client'
import { __awaiter } from '../../../../../_virtual/_tslib.js';
import { jsx } from 'react/jsx-runtime';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import '../../../components/Accordion/components/AccordionItem/AccordionItem.js';
import '../../../components/Alert/Alert.js';
import '../../../events/dynamicEvents.js';
import '@dynamic-labs/utils';
import '../../../context/DynamicContext/DynamicContext.js';
import '@dynamic-labs/sdk-api-core';
import '../../../shared/logger.js';
import '@dynamic-labs/iconic';
import '@dynamic-labs/wallet-connector-core';
import '../../../context/ViewContext/ViewContext.js';
import { ReactComponent as SvgAndroidTouchId } from '../../../shared/assets/android-touch-id.js';
import { ReactComponent as SvgMail } from '../../../shared/assets/mail.js';
import '@dynamic-labs/wallet-book';
import '../../../utils/constants/colors.js';
import '../../../utils/constants/values.js';
import '../../../store/state/loadingAndLifecycle.js';
import '../../../shared/consts/index.js';
import { useInternalDynamicContext } from '../../../context/DynamicContext/useDynamicContext/useInternalDynamicContext.js';
import '../../../context/CaptchaContext/CaptchaContext.js';
import '../../../context/ErrorContext/ErrorContext.js';
import '@dynamic-labs/multi-wallet';
import 'react-international-phone';
import '../../../config/ApiEndpoint.js';
import '../../../store/state/user/user.js';
import '../../../locale/locale.js';
import '../../../store/state/projectSettings/projectSettings.js';
import '../../../context/AccessDeniedContext/AccessDeniedContext.js';
import '../../../context/AccountExistsContext/AccountExistsContext.js';
import '../../../context/UserWalletsContext/UserWalletsContext.js';
import '../../../context/VerificationContext/VerificationContext.js';
import 'react-dom';
import { useElementById } from '../../../utils/hooks/useElementById/useElementById.js';
import '../../../context/WalletContext/WalletContext.js';
import { iframeContainerId } from '../PasskeyRecovery/constants.js';
import { useInitEmbeddedWalletAuthMethod } from '../../../utils/hooks/useInitEmbeddedWalletAuthMethod/useInitEmbeddedWalletAuthMethod.js';
import '../../../utils/hooks/useEmbeddedWallet/useSecureEnclaveEmbeddedWallet/constants.js';
import '../../../context/ThemeContext/ThemeContext.js';
import '../../../utils/hooks/useUserUpdateRequest/useUpdateUser/userFieldsSchema.js';
import '@dynamic-labs/types';
import '../../../context/LoadingContext/LoadingContext.js';
import 'yup';
import '../../../context/MockContext/MockContext.js';
import '../../CollectUserDataView/useFields.js';
import '../../../context/FieldsStateContext/FieldsStateContext.js';
import '../../../context/UserFieldEditorContext/UserFieldEditorContext.js';
import '@dynamic-labs/rpc-providers';
import '../../../store/state/environmentId.js';
import '../../../store/state/walletConnectorOptions.js';
import '../../../context/FooterAnimationContext/index.js';
import '../../../components/ShadowDOM/ShadowDOM.js';
import '../../../components/Transition/ZoomTransition/ZoomTransition.js';
import '../../../components/Transition/SlideInUpTransition/SlideInUpTransition.js';
import '../../../components/Transition/OpacityTransition/OpacityTransition.js';
import '../../../components/OverlayCard/OverlayCardTarget/OverlayCardTarget.js';
import '../../../context/SocialRedirectContext/SocialRedirectContext.js';
import '../../../context/WalletGroupContext/WalletGroupContext.js';
import '../../../widgets/DynamicWidget/components/DynamicWidgetHeader/DynamicWidgetHeader.js';
import 'react-focus-lock';
import { Icon } from '../../../components/Icon/Icon.js';
import '../../../widgets/DynamicWidget/context/DynamicWidgetContext.js';
import '../../../components/IconButton/IconButton.js';
import '../../../components/MenuList/Dropdown/Dropdown.js';
import 'formik';
import '../../../store/state/sendBalances.js';
import '../../../components/Input/Input.js';
import '../../TransactionConfirmationView/helpers/transactionErrorMessage.js';
import '../../../context/PasskeyContext/PasskeyContext.js';
import '../../../widgets/DynamicWidget/views/ManagePasskeysWidgetView/PasskeyCard/PasskeyCard.js';
import { PasskeyDeviceIcon } from '../PasskeyDeviceIcon/PasskeyDeviceIcon.js';
import '../../../../polyfills.js';
import '../../../context/ErrorBoundary/ErrorBoundaryBase.js';
import '../../../context/ErrorBoundary/ErrorBoundaryContext.js';
import '../../../widgets/DynamicBridgeWidget/context/DynamicBridgeWidgetContext/DynamicBridgeWidgetContext.js';
import '../../../widgets/DynamicWidget/components/DynamicWidgetCard/DynamicWidgetCard.js';
import '../../../components/IsBrowser/IsBrowser.js';
import '../../../context/WidgetRegistry/WidgetRegistryContextProvider.js';
import '../../../context/FundingContext/FundingContext.js';
import '../../../components/Popper/Popper/Popper.js';
import '../../../components/Popper/PopperContext/PopperContext.js';
import '../../../widgets/DynamicBridgeWidget/views/WalletsView/components/SecondaryWallets/SecondaryWallets.js';
import '@hcaptcha/react-hcaptcha';
import '../../../context/IpConfigurationContext/IpConfigurationContext.js';
import '../../../components/PasskeyCreatedSuccessBanner/PasskeyCreatedSuccessBanner.js';
import { ActionCard } from './ActionCard/ActionCard.js';
import '../../../context/SendBalanceContext/SendBalanceContext.js';
import '../../../context/ConnectWithOtpContext/ConnectWithOtpContext.js';
import '../../../context/ConnectWithOtpContext/constants.js';
import '../../../context/ReinitializeContext/ReinitializeContextProvider.js';
import '../../../store/state/tokenBalances.js';
import '../../../components/InlineWidget/InlineWidget.js';
import 'qrcode';

const EmbeddedAuthChoiceCards = ({ currentAction, supportedAuthenticators, }) => {
    var _a, _b;
    const [selectedAuthType, setSelectedAuthType] = useState();
    const { projectSettings } = useInternalDynamicContext();
    const { t } = useTranslation();
    const { initAuthMethod, isLoading } = useInitEmbeddedWalletAuthMethod();
    const { createRootElement } = useElementById();
    const iframeContainerRef = createRootElement(iframeContainerId);
    const supportedAuthenticatorsMap = {
        email: {
            badgeKey: 'dyn_embedded_authenticator.email.badge',
            description: t('dyn_embedded_authenticator.email.description', {
                expiration: (_b = (_a = projectSettings === null || projectSettings === void 0 ? void 0 : projectSettings.sdk.embeddedWallets) === null || _a === void 0 ? void 0 : _a.sessionKeyDuration) === null || _b === void 0 ? void 0 : _b.amount,
            }),
            descriptionKey: 'dyn_embedded_authenticator.email.description',
            icon: (jsx(Icon, { color: 'text-tertiary', size: 'small', children: jsx(SvgMail, {}) }, 'one-time-code-icon')),
            titleKey: 'dyn_embedded_authenticator.email.title',
        },
        passkey: {
            badgeKey: 'dyn_embedded_authenticator.passkey.badge',
            description: t('dyn_embedded_authenticator.passkey.description'),
            descriptionKey: 'dyn_embedded_authenticator.passkey.description',
            icon: (jsx(PasskeyDeviceIcon, { defaultIcon: jsx(SvgAndroidTouchId, { "data-testid": 'default-icon' }) })),
            titleKey: currentAction === 'reveal'
                ? 'dyn_embedded_authenticator.passkey.reveal_title'
                : 'dyn_embedded_authenticator.passkey.title',
        },
    };
    const handleSelection = (authenticatorType) => () => __awaiter(void 0, void 0, void 0, function* () {
        setSelectedAuthType(authenticatorType);
        yield initAuthMethod({
            authenticatorType: authenticatorType,
            currentAction,
            iframeContainer: iframeContainerRef.current,
        });
    });
    return (jsx("div", { className: 'embedded-auth-choice-cards', children: supportedAuthenticators &&
            supportedAuthenticators.map((authenticatorType) => (jsx(ActionCard, { LeadingIcon: supportedAuthenticatorsMap[authenticatorType]
                    .icon, action: handleSelection(authenticatorType), badgeKey: supportedAuthenticatorsMap[authenticatorType]
                    .badgeKey, badgeVariant: authenticatorType === 'passkey' ? 'primary' : 'secondary', descriptionKey: supportedAuthenticatorsMap[authenticatorType]
                    .descriptionKey, description: supportedAuthenticatorsMap[authenticatorType]
                    .description, titleKey: supportedAuthenticatorsMap[authenticatorType]
                    .titleKey, isLoading: isLoading && authenticatorType === selectedAuthType, dataTestId: `embedded-auth-choice__item-${authenticatorType}` }, authenticatorType))) }));
};

export { EmbeddedAuthChoiceCards };
