// @turnkey/http/dist/index.js
'use strict';

var public_api_fetcher = require('./__generated__/services/coordinator/public/v1/public_api.fetcher.js');
var public_api_client = require('./__generated__/services/coordinator/public/v1/public_api.client.js');
var config = require('./config.js');
var base = require('./base.js');
var shared = require('./shared.js');
var webauthn = require('./webauthn.js');
var async = require('./async.js');
var version = require('./version.js');

/**
 * @deprecated use `TurnkeyApi` instead
 */
const PublicApiService = public_api_fetcher;

exports.TurnkeyApi = public_api_fetcher;
exports.TurnkeyClient = public_api_client.TurnkeyClient;
exports.browserInit = config.browserInit;
exports.init = config.init;
exports.TurnkeyRequestError = base.TurnkeyRequestError;
exports.sealAndStampRequestBody = base.sealAndStampRequestBody;
exports.TurnkeyActivityError = shared.TurnkeyActivityError;
exports.getWebAuthnAttestation = webauthn.getWebAuthnAttestation;
exports.createActivityPoller = async.createActivityPoller;
exports.withAsyncPolling = async.withAsyncPolling;
exports.VERSION = version.VERSION;
exports.PublicApiService = PublicApiService;
