// @dynamic-labs/sdk-react-core/src/lib/shared/assets/shield.js
'use client'
import * as React from 'react';

var _rect, _path;
var _excluded = ["title", "titleId"];
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
var SvgShield = function SvgShield(_ref) {
  var title = _ref.title,
    titleId = _ref.titleId,
    props = _objectWithoutProperties(_ref, _excluded);
  return /*#__PURE__*/React.createElement("svg", _extends({
    width: 28,
    height: 28,
    viewBox: "0 0 28 28",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg",
    "aria-labelledby": titleId
  }, props), title ? /*#__PURE__*/React.createElement("title", {
    id: titleId
  }, title) : null, _rect || (_rect = /*#__PURE__*/React.createElement("rect", {
    width: 28,
    height: 28,
    rx: 14,
    fill: "#4779FF"
  })), _path || (_path = /*#__PURE__*/React.createElement("path", {
    fillRule: "evenodd",
    clipRule: "evenodd",
    d: "M13.444 5.88a.833.833 0 0 1 1.112 0A9.13 9.13 0 0 0 21.14 8.2c.395-.02.75.24.848.624.226.87.345 1.782.345 2.721 0 5.05-3.453 9.29-8.125 10.492a.833.833 0 0 1-.416 0c-4.672-1.202-8.125-5.443-8.125-10.492 0-.939.12-1.85.345-2.722a.833.833 0 0 1 .849-.623 9.13 9.13 0 0 0 6.584-2.322Zm-5.96 3.999c-.099.54-.15 1.097-.15 1.667A9.171 9.171 0 0 0 14 20.368a9.171 9.171 0 0 0 6.516-10.49A10.788 10.788 0 0 1 14 7.587a10.788 10.788 0 0 1-6.515 2.293Zm9.605 1.912a.833.833 0 0 1 0 1.178l-3.333 3.333a.833.833 0 0 1-1.178 0l-1.667-1.666a.834.834 0 0 1 1.178-1.179l1.078 1.078 2.744-2.745a.833.833 0 0 1 1.178 0Z",
    fill: "#fff"
  })));
};

export { SvgShield as ReactComponent };
