// @dynamic-labs/wallet-connector-core/src/index.js
'use client'
export { Chains, WalletConnectorBase, socialProviders } from './lib/WalletConnector.js';
export { Wallet } from './lib/wallets/Wallet/Wallet.js';
export { logger } from './utils/logger.js';
export { ProviderLookup } from './utils/ProviderLookup/ProviderLookup.js';
export { getChainInfo, getChainInfoWithOverrides, setChainInfoOverrides } from './utils/getChainInfo/getChainInfo.js';
export { isEmailOTPWalletConnector } from './utils/isEmailOTPWalletConnector/isEmailOTPWalletConnector.js';
export { isEmailWalletConnector } from './utils/isEmailWalletConnector/isEmailWalletConnector.js';
export { isTurnkeyWalletConnector } from './utils/isTurnkeyWalletConnector/isTurnkeyWalletConnector.js';
export { isSameAddress } from './utils/isSameAddress/isSameAddress.js';
export { isSocialWalletConnector } from './utils/isSocialWalletConnector/isSocialWalletConnector.js';
export { addHexPrefix, bufferToHex, utf8ToHex } from './utils/encoding.js';
export { getWalletConnectorByKey } from './utils/getWalletConnectorByKey.js';
export { shouldLowercaseAddress } from './utils/shouldLowercaseAddress.js';
export { getDeepLink } from './utils/walletConnectDeepLinks/walletConnectDeepLinks.js';
export { performPlatformSpecificConnectionMethod } from './utils/walletConnectDeepLinks/performPlatformSpecificConnectionMethod.js';
export { isEmbeddedConnector } from './utils/isEmbeddedConnector/isEmbeddedConnector.js';
export { isMagicConnector } from './utils/isMagicConnector/isMagicConnector.js';
export { isBloctoConnector } from './utils/isBloctoConnector/isBloctoConnector.js';
export { isPasskeyWalletConnector } from './utils/isPasskeyWalletConnector/isPasskeyWalletConnector.js';
export { isAccountAbstractionConnector } from './utils/isAccountAbstractionConnector/isAccountAbstractionConnector.js';
export { isBitcoinConnector } from './utils/isBitcoinConnector/isBitcoinConnector.js';
export { isHardwareWalletConnector } from './utils/isHardwareWalletConnector/isHardwareWalletConnector.js';
export { eventListenerHandlers } from './utils/eventListenerHandlers.js';
export { isHex } from './utils/isHex/isHex.js';
export { isPhantomRedirectConnector } from './utils/isPhantomRedirectConnector/isPhantomRedirectConnector.js';
export { getRpcUrlForChain } from './utils/getRpcUrlForChain/getRpcUrlForChain.js';
export { isCoinbaseMpcWalletConnector } from './utils/isCoinbaseMpcWalletConnector/isCoinbaseMpcWalletConnector.js';
export { isWalletConnectConnector } from './utils/isWalletConnectConnector/isWalletConnectConnector.js';
export { isSendBalanceWalletConnector } from './utils/isSendBalanceWalletConnector/isSendBalanceWalletConnector.js';
export { isSessionKeyCompatibleWallet } from './utils/isSessionKeyCompatible/isSessionKeyCompatibleWallet/isSessionKeyCompatibleWallet.js';
export { isSessionKeyCompatibleWalletConnector } from './utils/isSessionKeyCompatible/isSessionKeyCompatibleWalletConnector/isSessionKeyCompatibleWalletConnector.js';
export { getMobileExperience } from './utils/getMobileExperience/getMobileExperience.js';
