// @dynamic-labs/wallet-connector-core/src/index.cjs
'use client'
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var WalletConnector = require('./lib/WalletConnector.cjs');
var Wallet = require('./lib/wallets/Wallet/Wallet.cjs');
var logger = require('./utils/logger.cjs');
var ProviderLookup = require('./utils/ProviderLookup/ProviderLookup.cjs');
var getChainInfo = require('./utils/getChainInfo/getChainInfo.cjs');
var isEmailOTPWalletConnector = require('./utils/isEmailOTPWalletConnector/isEmailOTPWalletConnector.cjs');
var isEmailWalletConnector = require('./utils/isEmailWalletConnector/isEmailWalletConnector.cjs');
var isTurnkeyWalletConnector = require('./utils/isTurnkeyWalletConnector/isTurnkeyWalletConnector.cjs');
var isSameAddress = require('./utils/isSameAddress/isSameAddress.cjs');
var isSocialWalletConnector = require('./utils/isSocialWalletConnector/isSocialWalletConnector.cjs');
var encoding = require('./utils/encoding.cjs');
var getWalletConnectorByKey = require('./utils/getWalletConnectorByKey.cjs');
var shouldLowercaseAddress = require('./utils/shouldLowercaseAddress.cjs');
var walletConnectDeepLinks = require('./utils/walletConnectDeepLinks/walletConnectDeepLinks.cjs');
var performPlatformSpecificConnectionMethod = require('./utils/walletConnectDeepLinks/performPlatformSpecificConnectionMethod.cjs');
var isEmbeddedConnector = require('./utils/isEmbeddedConnector/isEmbeddedConnector.cjs');
var isMagicConnector = require('./utils/isMagicConnector/isMagicConnector.cjs');
var isBloctoConnector = require('./utils/isBloctoConnector/isBloctoConnector.cjs');
var isPasskeyWalletConnector = require('./utils/isPasskeyWalletConnector/isPasskeyWalletConnector.cjs');
var isAccountAbstractionConnector = require('./utils/isAccountAbstractionConnector/isAccountAbstractionConnector.cjs');
var isBitcoinConnector = require('./utils/isBitcoinConnector/isBitcoinConnector.cjs');
var isHardwareWalletConnector = require('./utils/isHardwareWalletConnector/isHardwareWalletConnector.cjs');
var eventListenerHandlers = require('./utils/eventListenerHandlers.cjs');
var isHex = require('./utils/isHex/isHex.cjs');
var isPhantomRedirectConnector = require('./utils/isPhantomRedirectConnector/isPhantomRedirectConnector.cjs');
var getRpcUrlForChain = require('./utils/getRpcUrlForChain/getRpcUrlForChain.cjs');
var isCoinbaseMpcWalletConnector = require('./utils/isCoinbaseMpcWalletConnector/isCoinbaseMpcWalletConnector.cjs');
var isWalletConnectConnector = require('./utils/isWalletConnectConnector/isWalletConnectConnector.cjs');
var isSendBalanceWalletConnector = require('./utils/isSendBalanceWalletConnector/isSendBalanceWalletConnector.cjs');
var isSessionKeyCompatibleWallet = require('./utils/isSessionKeyCompatible/isSessionKeyCompatibleWallet/isSessionKeyCompatibleWallet.cjs');
var isSessionKeyCompatibleWalletConnector = require('./utils/isSessionKeyCompatible/isSessionKeyCompatibleWalletConnector/isSessionKeyCompatibleWalletConnector.cjs');
var getMobileExperience = require('./utils/getMobileExperience/getMobileExperience.cjs');



exports.Chains = WalletConnector.Chains;
exports.WalletConnectorBase = WalletConnector.WalletConnectorBase;
exports.socialProviders = WalletConnector.socialProviders;
exports.Wallet = Wallet.Wallet;
exports.logger = logger.logger;
exports.ProviderLookup = ProviderLookup.ProviderLookup;
exports.getChainInfo = getChainInfo.getChainInfo;
exports.getChainInfoWithOverrides = getChainInfo.getChainInfoWithOverrides;
exports.setChainInfoOverrides = getChainInfo.setChainInfoOverrides;
exports.isEmailOTPWalletConnector = isEmailOTPWalletConnector.isEmailOTPWalletConnector;
exports.isEmailWalletConnector = isEmailWalletConnector.isEmailWalletConnector;
exports.isTurnkeyWalletConnector = isTurnkeyWalletConnector.isTurnkeyWalletConnector;
exports.isSameAddress = isSameAddress.isSameAddress;
exports.isSocialWalletConnector = isSocialWalletConnector.isSocialWalletConnector;
exports.addHexPrefix = encoding.addHexPrefix;
exports.bufferToHex = encoding.bufferToHex;
exports.utf8ToHex = encoding.utf8ToHex;
exports.getWalletConnectorByKey = getWalletConnectorByKey.getWalletConnectorByKey;
exports.shouldLowercaseAddress = shouldLowercaseAddress.shouldLowercaseAddress;
exports.getDeepLink = walletConnectDeepLinks.getDeepLink;
exports.performPlatformSpecificConnectionMethod = performPlatformSpecificConnectionMethod.performPlatformSpecificConnectionMethod;
exports.isEmbeddedConnector = isEmbeddedConnector.isEmbeddedConnector;
exports.isMagicConnector = isMagicConnector.isMagicConnector;
exports.isBloctoConnector = isBloctoConnector.isBloctoConnector;
exports.isPasskeyWalletConnector = isPasskeyWalletConnector.isPasskeyWalletConnector;
exports.isAccountAbstractionConnector = isAccountAbstractionConnector.isAccountAbstractionConnector;
exports.isBitcoinConnector = isBitcoinConnector.isBitcoinConnector;
exports.isHardwareWalletConnector = isHardwareWalletConnector.isHardwareWalletConnector;
exports.eventListenerHandlers = eventListenerHandlers.eventListenerHandlers;
exports.isHex = isHex.isHex;
exports.isPhantomRedirectConnector = isPhantomRedirectConnector.isPhantomRedirectConnector;
exports.getRpcUrlForChain = getRpcUrlForChain.getRpcUrlForChain;
exports.isCoinbaseMpcWalletConnector = isCoinbaseMpcWalletConnector.isCoinbaseMpcWalletConnector;
exports.isWalletConnectConnector = isWalletConnectConnector.isWalletConnectConnector;
exports.isSendBalanceWalletConnector = isSendBalanceWalletConnector.isSendBalanceWalletConnector;
exports.isSessionKeyCompatibleWallet = isSessionKeyCompatibleWallet.isSessionKeyCompatibleWallet;
exports.isSessionKeyCompatibleWalletConnector = isSessionKeyCompatibleWalletConnector.isSessionKeyCompatibleWalletConnector;
exports.getMobileExperience = getMobileExperience.getMobileExperience;
