// @dynamic-labs/sdk-react-core/src/lib/components/Transition/SlideInUpTransition/SlideInUpTransition.js
'use client'
import { __rest } from '../../../../../_virtual/_tslib.js';
import { jsx } from 'react/jsx-runtime';
import React__default from 'react';
import { useTransition } from '../../../utils/hooks/useTransition/useTransition.js';

const getSlideInUpTransitionDefaultStyles = (duration, curve = 'linear') => ({
    transition: `transform ${duration}ms ${curve}, opacity ${duration}ms ${curve}`,
});
const transitionStyles = {
    ENTERED: {
        transform: 'translateY(0)',
    },
    ENTERING: {
        transform: 'translateY(0.625rem)',
    },
    EXITING: {
        transform: 'translateY(0.625rem)',
    },
};
const UnoptimizedSlideInUpTransition = (_a) => {
    var { children, className } = _a, transitionProps = __rest(_a, ["children", "className"]);
    const { mount, stage, currentDuration } = useTransition(transitionProps);
    const defaultStyles = getSlideInUpTransitionDefaultStyles(currentDuration);
    const stageStyles = transitionStyles[stage];
    if (!mount)
        return null;
    return (jsx("div", { className: className, style: Object.assign(Object.assign({}, defaultStyles), stageStyles), children: children }));
};
const SlideInUpTransition = React__default.memo(UnoptimizedSlideInUpTransition);

export { SlideInUpTransition, UnoptimizedSlideInUpTransition };
