// @dynamic-labs/sdk-react-core/src/lib/context/DynamicContext/hooks/useRegisterStorageService/useRegisterStorageService.cjs
'use client'
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var utils = require('@dynamic-labs/utils');
var isSSR = require('../../../../shared/utils/functions/isSSR/isSSR.cjs');

/**
 * Registers the storage service for the SDK.
 * When on server side rendering, the storage service is not available, so we need to register a dummy storage service.
 * @param storagePostfix - The postfix for the storage services.
 */
const useRegisterStorageService = ({ storagePostfix, }) => {
    const storageForSSR = {
        clear: () => undefined,
        getItem: () => null,
        key: () => null,
        length: 0,
        removeItem: () => undefined,
        setItem: () => null,
    };
    const localStorage = isSSR.isSSR() ? storageForSSR : window.localStorage;
    const sessionStorage = isSSR.isSSR() ? storageForSSR : window.sessionStorage;
    if (!utils.StorageService.getSourceStorage('localStorage')) {
        utils.StorageService.registerSourceStorage('localStorage', utils.createStorageService({
            postfix: storagePostfix,
            storage: localStorage,
        }));
    }
    if (!utils.StorageService.getSourceStorage('sessionStorage')) {
        utils.StorageService.registerSourceStorage('sessionStorage', utils.createStorageService({
            postfix: storagePostfix,
            storage: sessionStorage,
        }));
    }
};

exports.useRegisterStorageService = useRegisterStorageService;
