// @dynamic-labs/sdk-api-core/src/models/Provider.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var runtime = require('../runtime.cjs');
var EcdsaValidatorOptions = require('./EcdsaValidatorOptions.cjs');
var ProviderAgreement = require('./ProviderAgreement.cjs');
var ProviderEnum = require('./ProviderEnum.cjs');
var SmsCountryCode = require('./SmsCountryCode.cjs');

/* tslint:disable */
/**
* @export
* @enum {string}
*/
exports.ProviderEntryPointVersionEnum = void 0;
(function (ProviderEntryPointVersionEnum) {
    ProviderEntryPointVersionEnum["V6"] = "v6";
    ProviderEntryPointVersionEnum["V7"] = "v7";
})(exports.ProviderEntryPointVersionEnum || (exports.ProviderEntryPointVersionEnum = {})); /**
* @export
* @enum {string}
*/
exports.ProviderKernelVersionEnum = void 0;
(function (ProviderKernelVersionEnum) {
    ProviderKernelVersionEnum["V24"] = "v2_4";
    ProviderKernelVersionEnum["V30"] = "v3_0";
    ProviderKernelVersionEnum["V31"] = "v3_1";
})(exports.ProviderKernelVersionEnum || (exports.ProviderKernelVersionEnum = {}));
function ProviderFromJSON(json) {
    return ProviderFromJSONTyped(json);
}
function ProviderFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !runtime.exists(json, 'id') ? undefined : json['id'],
        'provider': ProviderEnum.ProviderEnumFromJSON(json['provider']),
        'enabledAt': !runtime.exists(json, 'enabledAt') ? undefined : (json['enabledAt'] === null ? null : new Date(json['enabledAt'])),
        'clientId': !runtime.exists(json, 'clientId') ? undefined : json['clientId'],
        'clientSecret': !runtime.exists(json, 'clientSecret') ? undefined : json['clientSecret'],
        'providerProjectId': !runtime.exists(json, 'providerProjectId') ? undefined : json['providerProjectId'],
        'authorizationUrl': !runtime.exists(json, 'authorizationUrl') ? undefined : json['authorizationUrl'],
        'redirectUrl': !runtime.exists(json, 'redirectUrl') ? undefined : json['redirectUrl'],
        'defaultChainId': !runtime.exists(json, 'defaultChainId') ? undefined : json['defaultChainId'],
        'defaultChain': !runtime.exists(json, 'defaultChain') ? undefined : json['defaultChain'],
        'keyExportUrl': !runtime.exists(json, 'keyExportUrl') ? undefined : json['keyExportUrl'],
        'termsAcceptedByUser': !runtime.exists(json, 'termsAcceptedByUser') ? undefined : ProviderAgreement.ProviderAgreementFromJSON(json['termsAcceptedByUser']),
        'appleKeyId': !runtime.exists(json, 'appleKeyId') ? undefined : json['appleKeyId'],
        'appleTeamId': !runtime.exists(json, 'appleTeamId') ? undefined : json['appleTeamId'],
        'accountSid': !runtime.exists(json, 'accountSid') ? undefined : json['accountSid'],
        'twilioNumber': !runtime.exists(json, 'twilioNumber') ? undefined : json['twilioNumber'],
        'enabledCountries': !runtime.exists(json, 'enabledCountries') ? undefined : (json['enabledCountries'].map(SmsCountryCode.SmsCountryCodeFromJSON)),
        'entryPointVersion': !runtime.exists(json, 'entryPointVersion') ? undefined : json['entryPointVersion'],
        'kernelVersion': !runtime.exists(json, 'kernelVersion') ? undefined : json['kernelVersion'],
        'ecdsaProviderType': !runtime.exists(json, 'ecdsaProviderType') ? undefined : EcdsaValidatorOptions.EcdsaValidatorOptionsFromJSON(json['ecdsaProviderType']),
    };
}
function ProviderToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'provider': ProviderEnum.ProviderEnumToJSON(value.provider),
        'enabledAt': value.enabledAt === undefined ? undefined : (value.enabledAt === null ? null : value.enabledAt.toISOString()),
        'clientId': value.clientId,
        'clientSecret': value.clientSecret,
        'providerProjectId': value.providerProjectId,
        'authorizationUrl': value.authorizationUrl,
        'redirectUrl': value.redirectUrl,
        'defaultChainId': value.defaultChainId,
        'defaultChain': value.defaultChain,
        'keyExportUrl': value.keyExportUrl,
        'termsAcceptedByUser': ProviderAgreement.ProviderAgreementToJSON(value.termsAcceptedByUser),
        'appleKeyId': value.appleKeyId,
        'appleTeamId': value.appleTeamId,
        'accountSid': value.accountSid,
        'twilioNumber': value.twilioNumber,
        'enabledCountries': value.enabledCountries === undefined ? undefined : (value.enabledCountries.map(SmsCountryCode.SmsCountryCodeToJSON)),
        'entryPointVersion': value.entryPointVersion,
        'kernelVersion': value.kernelVersion,
        'ecdsaProviderType': EcdsaValidatorOptions.EcdsaValidatorOptionsToJSON(value.ecdsaProviderType),
    };
}

exports.ProviderFromJSON = ProviderFromJSON;
exports.ProviderFromJSONTyped = ProviderFromJSONTyped;
exports.ProviderToJSON = ProviderToJSON;
