// @dynamic-labs/sdk-react-core/src/lib/utils/constants/errors.cjs
'use client'
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

// HEADLESS ERRORS
const ACTIVE_SESSION = 'session_active';
const ERROR_SENDING_CODE = 'error_sending_code';
const ERROR_PROCESSING_REVEAL_REQUEST = 'error_processing_reveal_request';
const INVALID_REQUEST = 'invalid_request';
const INVALID_WALLET_DATA = 'invalid_wallet';
const SECURE_ENCLAVE_PROVIDER_NOT_ENABLED = 'secure_enclave_provider_not_enabled';
const USER_NOT_LOGGED_IN = 'user_not_logged_in';
const CODE_SENT = 'code_sent';
const SESSION_CREATED = 'session_created';
const SESSION_RESTORED = 'session_restored';
const ACCESS_BLOCKED_ERROR_CODE = 451;

exports.ACCESS_BLOCKED_ERROR_CODE = ACCESS_BLOCKED_ERROR_CODE;
exports.ACTIVE_SESSION = ACTIVE_SESSION;
exports.CODE_SENT = CODE_SENT;
exports.ERROR_PROCESSING_REVEAL_REQUEST = ERROR_PROCESSING_REVEAL_REQUEST;
exports.ERROR_SENDING_CODE = ERROR_SENDING_CODE;
exports.INVALID_REQUEST = INVALID_REQUEST;
exports.INVALID_WALLET_DATA = INVALID_WALLET_DATA;
exports.SECURE_ENCLAVE_PROVIDER_NOT_ENABLED = SECURE_ENCLAVE_PROVIDER_NOT_ENABLED;
exports.SESSION_CREATED = SESSION_CREATED;
exports.SESSION_RESTORED = SESSION_RESTORED;
exports.USER_NOT_LOGGED_IN = USER_NOT_LOGGED_IN;
