// @dynamic-labs/sdk-react-core/src/lib/utils/constants/localStorage.cjs
'use client'
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const AUTH_TOKEN = 'dynamic_authentication_token';
const AUTH_MIN_TOKEN = 'dynamic_min_authentication_token';
const AUTH_EXPIRES_AT = 'dynamic_auth_expires_at';
const LAST_USED_WALLET = 'dynamic_last_used_wallet';
const NONCE_STORAGE_KEY = 'dynamic_nonce';
const PRIMARY_WALLET_ID = 'dynamic_primary_wallet_id';
const WALLET_PICKER_SEARCH_KEY = 'dynamic_wallet_picker_search';
const NEWTOWEB3_WALLET_EXTENSION_FLAG_KEY = 'dynamic_newtoweb3_wallet_extension_installed';
const CONNECTED_WALLETS_INFO = 'dynamic_connected_wallets';
const AUTH_MODE = 'dynamic_auth_mode';
const CONNECTED_WALLET_NAME_SERVICE = 'dynamic_connected_wallet_ns';
const BRIDGE_ONBOARDING_COMPLETED = 'dynamic_bridge_onboarding_completed';
const DYNAMIC_CONTEXT_LAST_SESSION_SETTINGS = 'dynamic_context_session_settings';
const WAGMI_LAST_SESSION_SETTINGS = 'dynamic_wagmi_session_settings';
const EMBEDDED_WALLET_SESSION_SETTINGS = 'dynamic_embedded_wallet_session_settings';
const EMBEDDED_WALLET_SECURE_BANNER = 'dynamic_embedded_secure_banner';
const PHONE_INPUT_DEFAULT_COUNTRY = 'dynamic_phone_input_default_country';
const CAPTCHA_TOKEN = 'dynamic_captcha_token';
/**
 * Used to identify the purpose of a signature after phantom redirects back to the app in mobile
 */
const PHANTOM_SIGNATURE_STATE = 'phantom_verification_type';
const STORE_STORAGE_KEY = 'dynamic_store';

exports.AUTH_EXPIRES_AT = AUTH_EXPIRES_AT;
exports.AUTH_MIN_TOKEN = AUTH_MIN_TOKEN;
exports.AUTH_MODE = AUTH_MODE;
exports.AUTH_TOKEN = AUTH_TOKEN;
exports.BRIDGE_ONBOARDING_COMPLETED = BRIDGE_ONBOARDING_COMPLETED;
exports.CAPTCHA_TOKEN = CAPTCHA_TOKEN;
exports.CONNECTED_WALLETS_INFO = CONNECTED_WALLETS_INFO;
exports.CONNECTED_WALLET_NAME_SERVICE = CONNECTED_WALLET_NAME_SERVICE;
exports.DYNAMIC_CONTEXT_LAST_SESSION_SETTINGS = DYNAMIC_CONTEXT_LAST_SESSION_SETTINGS;
exports.EMBEDDED_WALLET_SECURE_BANNER = EMBEDDED_WALLET_SECURE_BANNER;
exports.EMBEDDED_WALLET_SESSION_SETTINGS = EMBEDDED_WALLET_SESSION_SETTINGS;
exports.LAST_USED_WALLET = LAST_USED_WALLET;
exports.NEWTOWEB3_WALLET_EXTENSION_FLAG_KEY = NEWTOWEB3_WALLET_EXTENSION_FLAG_KEY;
exports.NONCE_STORAGE_KEY = NONCE_STORAGE_KEY;
exports.PHANTOM_SIGNATURE_STATE = PHANTOM_SIGNATURE_STATE;
exports.PHONE_INPUT_DEFAULT_COUNTRY = PHONE_INPUT_DEFAULT_COUNTRY;
exports.PRIMARY_WALLET_ID = PRIMARY_WALLET_ID;
exports.STORE_STORAGE_KEY = STORE_STORAGE_KEY;
exports.WAGMI_LAST_SESSION_SETTINGS = WAGMI_LAST_SESSION_SETTINGS;
exports.WALLET_PICKER_SEARCH_KEY = WALLET_PICKER_SEARCH_KEY;
