// @dynamic-labs/sdk-api-core/src/index.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var runtime = require('./runtime.cjs');
var SDKApi = require('./apis/SDKApi.cjs');
var AttestationConveyancePreference = require('./models/AttestationConveyancePreference.cjs');
var AuthModeEnum = require('./models/AuthModeEnum.cjs');
var AuthSettings = require('./models/AuthSettings.cjs');
var AuthStorageEnum = require('./models/AuthStorageEnum.cjs');
var AuthenticationExtensionsClientInputs = require('./models/AuthenticationExtensionsClientInputs.cjs');
var AuthenticationExtensionsClientOutputs = require('./models/AuthenticationExtensionsClientOutputs.cjs');
var AuthenticatorAssertionResponse = require('./models/AuthenticatorAssertionResponse.cjs');
var AuthenticatorAttachment = require('./models/AuthenticatorAttachment.cjs');
var AuthenticatorAttestationResponse = require('./models/AuthenticatorAttestationResponse.cjs');
var AuthenticatorSelectionCriteria = require('./models/AuthenticatorSelectionCriteria.cjs');
var AuthenticatorTransportProtocol = require('./models/AuthenticatorTransportProtocol.cjs');
var BadRequest = require('./models/BadRequest.cjs');
var BaseUser = require('./models/BaseUser.cjs');
var ChainConfiguration = require('./models/ChainConfiguration.cjs');
var ChainEnum = require('./models/ChainEnum.cjs');
var CoinbaseMpcWalletProperties = require('./models/CoinbaseMpcWalletProperties.cjs');
var CompletePasskeyRecoveryRequest = require('./models/CompletePasskeyRecoveryRequest.cjs');
var ConnectRequest = require('./models/ConnectRequest.cjs');
var CreateEmbeddedWalletParams = require('./models/CreateEmbeddedWalletParams.cjs');
var CreateEmbeddedWalletSpecificOpts = require('./models/CreateEmbeddedWalletSpecificOpts.cjs');
var CreateEmbeddedWalletsRequest = require('./models/CreateEmbeddedWalletsRequest.cjs');
var CreateMfaToken = require('./models/CreateMfaToken.cjs');
var CreateTurnkeyEmbeddedWalletSpecificOpts = require('./models/CreateTurnkeyEmbeddedWalletSpecificOpts.cjs');
var CreateUserEmbeddedWalletsFromFarcasterRequest = require('./models/CreateUserEmbeddedWalletsFromFarcasterRequest.cjs');
var CreateUserEmbeddedWalletsRequest = require('./models/CreateUserEmbeddedWalletsRequest.cjs');
var CreateWalletAccountRequest = require('./models/CreateWalletAccountRequest.cjs');
var Currency = require('./models/Currency.cjs');
var CurrencyType = require('./models/CurrencyType.cjs');
var CustomFieldType = require('./models/CustomFieldType.cjs');
var CustomFieldValidValue = require('./models/CustomFieldValidValue.cjs');
var CustomFieldValidationRules = require('./models/CustomFieldValidationRules.cjs');
var Duration = require('./models/Duration.cjs');
var DynamicJwt = require('./models/DynamicJwt.cjs');
var EcdsaValidatorOptions = require('./models/EcdsaValidatorOptions.cjs');
var EmailProviderResponse = require('./models/EmailProviderResponse.cjs');
var EmailVerificationCreateRequest = require('./models/EmailVerificationCreateRequest.cjs');
var EmailVerificationCreateResponse = require('./models/EmailVerificationCreateResponse.cjs');
var EmailVerificationRetryRequest = require('./models/EmailVerificationRetryRequest.cjs');
var EmailVerificationVerifyRequest = require('./models/EmailVerificationVerifyRequest.cjs');
var EmbeddedWalletAuthToken = require('./models/EmbeddedWalletAuthToken.cjs');
var EmbeddedWalletAuthType = require('./models/EmbeddedWalletAuthType.cjs');
var EmbeddedWalletChainEnum = require('./models/EmbeddedWalletChainEnum.cjs');
var EmbeddedWalletPasscodeClaimRequest = require('./models/EmbeddedWalletPasscodeClaimRequest.cjs');
var EmbeddedWalletProviderEnum = require('./models/EmbeddedWalletProviderEnum.cjs');
var EmbeddedWalletSecret = require('./models/EmbeddedWalletSecret.cjs');
var EmbeddedWalletSecretWithUpdatedJwt = require('./models/EmbeddedWalletSecretWithUpdatedJwt.cjs');
var EmbeddedWalletSecretWithUpdatedJwtAllOf = require('./models/EmbeddedWalletSecretWithUpdatedJwtAllOf.cjs');
var EmbeddedWalletSecurityMethod = require('./models/EmbeddedWalletSecurityMethod.cjs');
var EmbeddedWalletVersionEnum = require('./models/EmbeddedWalletVersionEnum.cjs');
var EnvironmentEnum = require('./models/EnvironmentEnum.cjs');
var ErrorMessageWithCode = require('./models/ErrorMessageWithCode.cjs');
var ExportEmbeddedWalletResponse = require('./models/ExportEmbeddedWalletResponse.cjs');
var ExternalAuth = require('./models/ExternalAuth.cjs');
var ExternalAuthSigninRequest = require('./models/ExternalAuthSigninRequest.cjs');
var FarcasterSignInRequest = require('./models/FarcasterSignInRequest.cjs');
var Forbidden = require('./models/Forbidden.cjs');
var ForbiddenErrorPayload = require('./models/ForbiddenErrorPayload.cjs');
var ForbiddenWithErrorAndPayload = require('./models/ForbiddenWithErrorAndPayload.cjs');
var FrameworkSettings = require('./models/FrameworkSettings.cjs');
var GetUserPasskeysResponse = require('./models/GetUserPasskeysResponse.cjs');
var HCaptchaSettings = require('./models/HCaptchaSettings.cjs');
var HardwareWalletEnum = require('./models/HardwareWalletEnum.cjs');
var HardwareWalletProperties = require('./models/HardwareWalletProperties.cjs');
var HealthcheckResponse = require('./models/HealthcheckResponse.cjs');
var HealthcheckStatus = require('./models/HealthcheckStatus.cjs');
var InitEmailAuthRequest = require('./models/InitEmailAuthRequest.cjs');
var InitEmailAuthResponse = require('./models/InitEmailAuthResponse.cjs');
var InitPasskeyRecoveryRequest = require('./models/InitPasskeyRecoveryRequest.cjs');
var InitPasskeyRecoveryResponse = require('./models/InitPasskeyRecoveryResponse.cjs');
var IntegrationSetting = require('./models/IntegrationSetting.cjs');
var InternalServerError = require('./models/InternalServerError.cjs');
var JwksKey = require('./models/JwksKey.cjs');
var JwksResponse = require('./models/JwksResponse.cjs');
var JwtBlockchainAccount = require('./models/JwtBlockchainAccount.cjs');
var JwtPayloadDeprecatedInfo = require('./models/JwtPayloadDeprecatedInfo.cjs');
var JwtVerifiedCredential = require('./models/JwtVerifiedCredential.cjs');
var JwtVerifiedCredentialFormatEnum = require('./models/JwtVerifiedCredentialFormatEnum.cjs');
var KycFieldType = require('./models/KycFieldType.cjs');
var MFAAuthPasskeyDeviceGetResponse = require('./models/MFAAuthPasskeyDeviceGetResponse.cjs');
var MFAAuthPasskeyDeviceGetResponseAssertion = require('./models/MFAAuthPasskeyDeviceGetResponseAssertion.cjs');
var MFAAuthPasskeyDevicePostRequest = require('./models/MFAAuthPasskeyDevicePostRequest.cjs');
var MFAAuthRecoveryDevicePostRequest = require('./models/MFAAuthRecoveryDevicePostRequest.cjs');
var MFAAuthTotpDevicePostRequest = require('./models/MFAAuthTotpDevicePostRequest.cjs');
var MFADevice = require('./models/MFADevice.cjs');
var MFADeviceType = require('./models/MFADeviceType.cjs');
var MFAGetRecoveryCodesResponse = require('./models/MFAGetRecoveryCodesResponse.cjs');
var MFAListDevicesResponse = require('./models/MFAListDevicesResponse.cjs');
var MFARegenRecoveryCodesResponse = require('./models/MFARegenRecoveryCodesResponse.cjs');
var MFARegisterPasskeyDeviceGetResponse = require('./models/MFARegisterPasskeyDeviceGetResponse.cjs');
var MFARegisterPasskeyDeviceGetResponseRegistration = require('./models/MFARegisterPasskeyDeviceGetResponseRegistration.cjs');
var MFARegisterPasskeyDevicePostRequest = require('./models/MFARegisterPasskeyDevicePostRequest.cjs');
var MFARegisterTotpDeviceGetResponse = require('./models/MFARegisterTotpDeviceGetResponse.cjs');
var MFARegisterTotpDevicePostRequest = require('./models/MFARegisterTotpDevicePostRequest.cjs');
var MFASettings = require('./models/MFASettings.cjs');
var MFAUpdateDeviceRequest = require('./models/MFAUpdateDeviceRequest.cjs');
var MergeConflicts = require('./models/MergeConflicts.cjs');
var MergeUser = require('./models/MergeUser.cjs');
var MergeUserConflict = require('./models/MergeUserConflict.cjs');
var MergeUserConflictResolution = require('./models/MergeUserConflictResolution.cjs');
var MergeUserConflictResolutions = require('./models/MergeUserConflictResolutions.cjs');
var MethodNotAllowed = require('./models/MethodNotAllowed.cjs');
var MfaBackupCodeAcknowledgement = require('./models/MfaBackupCodeAcknowledgement.cjs');
var MinifiedDynamicJwt = require('./models/MinifiedDynamicJwt.cjs');
var MobileSettings = require('./models/MobileSettings.cjs');
var NameService = require('./models/NameService.cjs');
var NameServiceData = require('./models/NameServiceData.cjs');
var NativeCurrency = require('./models/NativeCurrency.cjs');
var Network = require('./models/Network.cjs');
var NetworkConfiguration = require('./models/NetworkConfiguration.cjs');
var NetworkConfigurationResponse = require('./models/NetworkConfigurationResponse.cjs');
var NextJsSettings = require('./models/NextJsSettings.cjs');
var NextViewEnum = require('./models/NextViewEnum.cjs');
var NonceResponse = require('./models/NonceResponse.cjs');
var NotFound = require('./models/NotFound.cjs');
var OAuthAccount = require('./models/OAuthAccount.cjs');
var OauthInitAuthRequest = require('./models/OauthInitAuthRequest.cjs');
var OauthProviderLoginUrl = require('./models/OauthProviderLoginUrl.cjs');
var OauthProviderRequest = require('./models/OauthProviderRequest.cjs');
var OauthRequest = require('./models/OauthRequest.cjs');
var OauthResultRequest = require('./models/OauthResultRequest.cjs');
var OauthResultResponse = require('./models/OauthResultResponse.cjs');
var OauthResultStatus = require('./models/OauthResultStatus.cjs');
var OnrampConfiguration = require('./models/OnrampConfiguration.cjs');
var PasskeyAuthRequest = require('./models/PasskeyAuthRequest.cjs');
var PasskeyRegisterVerifyRequest = require('./models/PasskeyRegisterVerifyRequest.cjs');
var PasskeyRegistrationCredential = require('./models/PasskeyRegistrationCredential.cjs');
var PasskeyStorage = require('./models/PasskeyStorage.cjs');
var PasswordSourceTypeEnum = require('./models/PasswordSourceTypeEnum.cjs');
var PrefetchRequest = require('./models/PrefetchRequest.cjs');
var ProjectSettings = require('./models/ProjectSettings.cjs');
var ProjectSettingsChains = require('./models/ProjectSettingsChains.cjs');
var ProjectSettingsDesign = require('./models/ProjectSettingsDesign.cjs');
var ProjectSettingsDesignButton = require('./models/ProjectSettingsDesignButton.cjs');
var ProjectSettingsDesignModal = require('./models/ProjectSettingsDesignModal.cjs');
var ProjectSettingsDesignWidget = require('./models/ProjectSettingsDesignWidget.cjs');
var ProjectSettingsGeneral = require('./models/ProjectSettingsGeneral.cjs');
var ProjectSettingsKyc = require('./models/ProjectSettingsKyc.cjs');
var ProjectSettingsPrivacy = require('./models/ProjectSettingsPrivacy.cjs');
var ProjectSettingsSdk = require('./models/ProjectSettingsSdk.cjs');
var ProjectSettingsSdkAccountAbstraction = require('./models/ProjectSettingsSdkAccountAbstraction.cjs');
var ProjectSettingsSdkEmailSignIn = require('./models/ProjectSettingsSdkEmailSignIn.cjs');
var ProjectSettingsSdkEmbeddedWallets = require('./models/ProjectSettingsSdkEmbeddedWallets.cjs');
var ProjectSettingsSdkSocialSignIn = require('./models/ProjectSettingsSdkSocialSignIn.cjs');
var ProjectSettingsSdkWalletConnect = require('./models/ProjectSettingsSdkWalletConnect.cjs');
var ProjectSettingsSecurity = require('./models/ProjectSettingsSecurity.cjs');
var Provider = require('./models/Provider.cjs');
var ProviderAgreement = require('./models/ProviderAgreement.cjs');
var ProviderEnum = require('./models/ProviderEnum.cjs');
var PublicKeyCredentialDescriptor = require('./models/PublicKeyCredentialDescriptor.cjs');
var PublicKeyCredentialParameters = require('./models/PublicKeyCredentialParameters.cjs');
var PublicKeyCredentialRpEntity = require('./models/PublicKeyCredentialRpEntity.cjs');
var PublicKeyCredentialType = require('./models/PublicKeyCredentialType.cjs');
var PublicKeyCredentialUserEntity = require('./models/PublicKeyCredentialUserEntity.cjs');
var PublishEvents = require('./models/PublishEvents.cjs');
var PublishEventsEvents = require('./models/PublishEventsEvents.cjs');
var ReactSettings = require('./models/ReactSettings.cjs');
var RegisterEmbeddedWalletSessionKeyResponse = require('./models/RegisterEmbeddedWalletSessionKeyResponse.cjs');
var RegisterSessionKeyRequest = require('./models/RegisterSessionKeyRequest.cjs');
var ResidentKeyRequirement = require('./models/ResidentKeyRequirement.cjs');
var ScanWebsiteUrlRequest = require('./models/ScanWebsiteUrlRequest.cjs');
var ScanWebsiteUrlResponse = require('./models/ScanWebsiteUrlResponse.cjs');
var SdkSettingsRequest = require('./models/SdkSettingsRequest.cjs');
var SdkUser = require('./models/SdkUser.cjs');
var SdkView = require('./models/SdkView.cjs');
var SdkViewSection = require('./models/SdkViewSection.cjs');
var SdkViewSectionAlignment = require('./models/SdkViewSectionAlignment.cjs');
var SdkViewSectionType = require('./models/SdkViewSectionType.cjs');
var SdkViewType = require('./models/SdkViewType.cjs');
var SdkViewUpdateRequest = require('./models/SdkViewUpdateRequest.cjs');
var SdkViewsResponse = require('./models/SdkViewsResponse.cjs');
var SignInProviderEnum = require('./models/SignInProviderEnum.cjs');
var SmsCountryCode = require('./models/SmsCountryCode.cjs');
var SmsVerificationCreateRequest = require('./models/SmsVerificationCreateRequest.cjs');
var SmsVerificationCreateResponse = require('./models/SmsVerificationCreateResponse.cjs');
var SmsVerificationRetryRequest = require('./models/SmsVerificationRetryRequest.cjs');
var SmsVerificationVerifyRequest = require('./models/SmsVerificationVerifyRequest.cjs');
var SocialSignInProvider = require('./models/SocialSignInProvider.cjs');
var SocialSignInProviderEnum = require('./models/SocialSignInProviderEnum.cjs');
var SolanaTransactionOptimizationRequest = require('./models/SolanaTransactionOptimizationRequest.cjs');
var SolanaTransactionOptimizationResponse = require('./models/SolanaTransactionOptimizationResponse.cjs');
var SupportedOnrampsResponse = require('./models/SupportedOnrampsResponse.cjs');
var SupportedSecurityMethod = require('./models/SupportedSecurityMethod.cjs');
var SupportedSecurityMethods = require('./models/SupportedSecurityMethods.cjs');
var TelegramPostRequest = require('./models/TelegramPostRequest.cjs');
var TelegramUser = require('./models/TelegramUser.cjs');
var TimeUnitEnum = require('./models/TimeUnitEnum.cjs');
var TokenBalance = require('./models/TokenBalance.cjs');
var TooManyRequests = require('./models/TooManyRequests.cjs');
var TurnkeyCreateWalletAccountsRequestBody = require('./models/TurnkeyCreateWalletAccountsRequestBody.cjs');
var TurnkeyCreateWalletAccountsRequestBodyParameters = require('./models/TurnkeyCreateWalletAccountsRequestBodyParameters.cjs');
var TurnkeySignedRequest = require('./models/TurnkeySignedRequest.cjs');
var TurnkeyStamp = require('./models/TurnkeyStamp.cjs');
var TurnkeyWalletAccount = require('./models/TurnkeyWalletAccount.cjs');
var TurnkeyWalletProperties = require('./models/TurnkeyWalletProperties.cjs');
var Unauthorized = require('./models/Unauthorized.cjs');
var UnprocessableEntity = require('./models/UnprocessableEntity.cjs');
var UnprocessableEntityErrorCode = require('./models/UnprocessableEntityErrorCode.cjs');
var UnprocessableEntityErrorPayload = require('./models/UnprocessableEntityErrorPayload.cjs');
var UpdateRecoveryEmailRequest = require('./models/UpdateRecoveryEmailRequest.cjs');
var UpdateSelfResponse = require('./models/UpdateSelfResponse.cjs');
var UpdateSelfResponseAllOf = require('./models/UpdateSelfResponseAllOf.cjs');
var UpdateUserPasskeyRequest = require('./models/UpdateUserPasskeyRequest.cjs');
var UserFields = require('./models/UserFields.cjs');
var UserIdentifierTypeEnum = require('./models/UserIdentifierTypeEnum.cjs');
var UserOauthAccessTokenResponse = require('./models/UserOauthAccessTokenResponse.cjs');
var UserPasskey = require('./models/UserPasskey.cjs');
var UserVerificationRequirement = require('./models/UserVerificationRequirement.cjs');
var UserWalletSelectionRequest = require('./models/UserWalletSelectionRequest.cjs');
var VerifyRequest = require('./models/VerifyRequest.cjs');
var VerifyResponse = require('./models/VerifyResponse.cjs');
var VerifyUnlinkRequest = require('./models/VerifyUnlinkRequest.cjs');
var WalletAdditionalAddress = require('./models/WalletAdditionalAddress.cjs');
var WalletAddressType = require('./models/WalletAddressType.cjs');
var WalletProperties = require('./models/WalletProperties.cjs');
var WalletProviderEnum = require('./models/WalletProviderEnum.cjs');



exports.BASE_PATH = runtime.BASE_PATH;
exports.BaseAPI = runtime.BaseAPI;
exports.BlobApiResponse = runtime.BlobApiResponse;
exports.COLLECTION_FORMATS = runtime.COLLECTION_FORMATS;
exports.Configuration = runtime.Configuration;
exports.JSONApiResponse = runtime.JSONApiResponse;
exports.RequiredError = runtime.RequiredError;
exports.TextApiResponse = runtime.TextApiResponse;
exports.VoidApiResponse = runtime.VoidApiResponse;
exports.canConsumeForm = runtime.canConsumeForm;
exports.exists = runtime.exists;
exports.mapValues = runtime.mapValues;
exports.querystring = runtime.querystring;
Object.defineProperty(exports, 'GetAccountBalancesChainNameEnum', {
	enumerable: true,
	get: function () { return SDKApi.GetAccountBalancesChainNameEnum; }
});
Object.defineProperty(exports, 'GetAccountBalancesNetworkIdEnum', {
	enumerable: true,
	get: function () { return SDKApi.GetAccountBalancesNetworkIdEnum; }
});
Object.defineProperty(exports, 'GetAccountBalancesOptionsChainNameEnum', {
	enumerable: true,
	get: function () { return SDKApi.GetAccountBalancesOptionsChainNameEnum; }
});
exports.SDKApi = SDKApi.SDKApi;
Object.defineProperty(exports, 'AttestationConveyancePreference', {
	enumerable: true,
	get: function () { return AttestationConveyancePreference.AttestationConveyancePreference; }
});
exports.AttestationConveyancePreferenceFromJSON = AttestationConveyancePreference.AttestationConveyancePreferenceFromJSON;
exports.AttestationConveyancePreferenceFromJSONTyped = AttestationConveyancePreference.AttestationConveyancePreferenceFromJSONTyped;
exports.AttestationConveyancePreferenceToJSON = AttestationConveyancePreference.AttestationConveyancePreferenceToJSON;
Object.defineProperty(exports, 'AuthModeEnum', {
	enumerable: true,
	get: function () { return AuthModeEnum.AuthModeEnum; }
});
exports.AuthModeEnumFromJSON = AuthModeEnum.AuthModeEnumFromJSON;
exports.AuthModeEnumFromJSONTyped = AuthModeEnum.AuthModeEnumFromJSONTyped;
exports.AuthModeEnumToJSON = AuthModeEnum.AuthModeEnumToJSON;
exports.AuthSettingsFromJSON = AuthSettings.AuthSettingsFromJSON;
exports.AuthSettingsFromJSONTyped = AuthSettings.AuthSettingsFromJSONTyped;
exports.AuthSettingsToJSON = AuthSettings.AuthSettingsToJSON;
Object.defineProperty(exports, 'AuthStorageEnum', {
	enumerable: true,
	get: function () { return AuthStorageEnum.AuthStorageEnum; }
});
exports.AuthStorageEnumFromJSON = AuthStorageEnum.AuthStorageEnumFromJSON;
exports.AuthStorageEnumFromJSONTyped = AuthStorageEnum.AuthStorageEnumFromJSONTyped;
exports.AuthStorageEnumToJSON = AuthStorageEnum.AuthStorageEnumToJSON;
exports.AuthenticationExtensionsClientInputsFromJSON = AuthenticationExtensionsClientInputs.AuthenticationExtensionsClientInputsFromJSON;
exports.AuthenticationExtensionsClientInputsFromJSONTyped = AuthenticationExtensionsClientInputs.AuthenticationExtensionsClientInputsFromJSONTyped;
exports.AuthenticationExtensionsClientInputsToJSON = AuthenticationExtensionsClientInputs.AuthenticationExtensionsClientInputsToJSON;
exports.AuthenticationExtensionsClientOutputsFromJSON = AuthenticationExtensionsClientOutputs.AuthenticationExtensionsClientOutputsFromJSON;
exports.AuthenticationExtensionsClientOutputsFromJSONTyped = AuthenticationExtensionsClientOutputs.AuthenticationExtensionsClientOutputsFromJSONTyped;
exports.AuthenticationExtensionsClientOutputsToJSON = AuthenticationExtensionsClientOutputs.AuthenticationExtensionsClientOutputsToJSON;
exports.AuthenticatorAssertionResponseFromJSON = AuthenticatorAssertionResponse.AuthenticatorAssertionResponseFromJSON;
exports.AuthenticatorAssertionResponseFromJSONTyped = AuthenticatorAssertionResponse.AuthenticatorAssertionResponseFromJSONTyped;
exports.AuthenticatorAssertionResponseToJSON = AuthenticatorAssertionResponse.AuthenticatorAssertionResponseToJSON;
Object.defineProperty(exports, 'AuthenticatorAttachment', {
	enumerable: true,
	get: function () { return AuthenticatorAttachment.AuthenticatorAttachment; }
});
exports.AuthenticatorAttachmentFromJSON = AuthenticatorAttachment.AuthenticatorAttachmentFromJSON;
exports.AuthenticatorAttachmentFromJSONTyped = AuthenticatorAttachment.AuthenticatorAttachmentFromJSONTyped;
exports.AuthenticatorAttachmentToJSON = AuthenticatorAttachment.AuthenticatorAttachmentToJSON;
exports.AuthenticatorAttestationResponseFromJSON = AuthenticatorAttestationResponse.AuthenticatorAttestationResponseFromJSON;
exports.AuthenticatorAttestationResponseFromJSONTyped = AuthenticatorAttestationResponse.AuthenticatorAttestationResponseFromJSONTyped;
exports.AuthenticatorAttestationResponseToJSON = AuthenticatorAttestationResponse.AuthenticatorAttestationResponseToJSON;
exports.AuthenticatorSelectionCriteriaFromJSON = AuthenticatorSelectionCriteria.AuthenticatorSelectionCriteriaFromJSON;
exports.AuthenticatorSelectionCriteriaFromJSONTyped = AuthenticatorSelectionCriteria.AuthenticatorSelectionCriteriaFromJSONTyped;
exports.AuthenticatorSelectionCriteriaToJSON = AuthenticatorSelectionCriteria.AuthenticatorSelectionCriteriaToJSON;
Object.defineProperty(exports, 'AuthenticatorTransportProtocol', {
	enumerable: true,
	get: function () { return AuthenticatorTransportProtocol.AuthenticatorTransportProtocol; }
});
exports.AuthenticatorTransportProtocolFromJSON = AuthenticatorTransportProtocol.AuthenticatorTransportProtocolFromJSON;
exports.AuthenticatorTransportProtocolFromJSONTyped = AuthenticatorTransportProtocol.AuthenticatorTransportProtocolFromJSONTyped;
exports.AuthenticatorTransportProtocolToJSON = AuthenticatorTransportProtocol.AuthenticatorTransportProtocolToJSON;
exports.BadRequestFromJSON = BadRequest.BadRequestFromJSON;
exports.BadRequestFromJSONTyped = BadRequest.BadRequestFromJSONTyped;
exports.BadRequestToJSON = BadRequest.BadRequestToJSON;
exports.BaseUserFromJSON = BaseUser.BaseUserFromJSON;
exports.BaseUserFromJSONTyped = BaseUser.BaseUserFromJSONTyped;
exports.BaseUserToJSON = BaseUser.BaseUserToJSON;
exports.ChainConfigurationFromJSON = ChainConfiguration.ChainConfigurationFromJSON;
exports.ChainConfigurationFromJSONTyped = ChainConfiguration.ChainConfigurationFromJSONTyped;
exports.ChainConfigurationToJSON = ChainConfiguration.ChainConfigurationToJSON;
Object.defineProperty(exports, 'ChainEnum', {
	enumerable: true,
	get: function () { return ChainEnum.ChainEnum; }
});
exports.ChainEnumFromJSON = ChainEnum.ChainEnumFromJSON;
exports.ChainEnumFromJSONTyped = ChainEnum.ChainEnumFromJSONTyped;
exports.ChainEnumToJSON = ChainEnum.ChainEnumToJSON;
exports.CoinbaseMpcWalletPropertiesFromJSON = CoinbaseMpcWalletProperties.CoinbaseMpcWalletPropertiesFromJSON;
exports.CoinbaseMpcWalletPropertiesFromJSONTyped = CoinbaseMpcWalletProperties.CoinbaseMpcWalletPropertiesFromJSONTyped;
exports.CoinbaseMpcWalletPropertiesToJSON = CoinbaseMpcWalletProperties.CoinbaseMpcWalletPropertiesToJSON;
exports.CompletePasskeyRecoveryRequestFromJSON = CompletePasskeyRecoveryRequest.CompletePasskeyRecoveryRequestFromJSON;
exports.CompletePasskeyRecoveryRequestFromJSONTyped = CompletePasskeyRecoveryRequest.CompletePasskeyRecoveryRequestFromJSONTyped;
exports.CompletePasskeyRecoveryRequestToJSON = CompletePasskeyRecoveryRequest.CompletePasskeyRecoveryRequestToJSON;
exports.ConnectRequestFromJSON = ConnectRequest.ConnectRequestFromJSON;
exports.ConnectRequestFromJSONTyped = ConnectRequest.ConnectRequestFromJSONTyped;
exports.ConnectRequestToJSON = ConnectRequest.ConnectRequestToJSON;
exports.CreateEmbeddedWalletParamsFromJSON = CreateEmbeddedWalletParams.CreateEmbeddedWalletParamsFromJSON;
exports.CreateEmbeddedWalletParamsFromJSONTyped = CreateEmbeddedWalletParams.CreateEmbeddedWalletParamsFromJSONTyped;
exports.CreateEmbeddedWalletParamsToJSON = CreateEmbeddedWalletParams.CreateEmbeddedWalletParamsToJSON;
exports.CreateEmbeddedWalletSpecificOptsFromJSON = CreateEmbeddedWalletSpecificOpts.CreateEmbeddedWalletSpecificOptsFromJSON;
exports.CreateEmbeddedWalletSpecificOptsFromJSONTyped = CreateEmbeddedWalletSpecificOpts.CreateEmbeddedWalletSpecificOptsFromJSONTyped;
exports.CreateEmbeddedWalletSpecificOptsToJSON = CreateEmbeddedWalletSpecificOpts.CreateEmbeddedWalletSpecificOptsToJSON;
exports.CreateEmbeddedWalletsRequestFromJSON = CreateEmbeddedWalletsRequest.CreateEmbeddedWalletsRequestFromJSON;
exports.CreateEmbeddedWalletsRequestFromJSONTyped = CreateEmbeddedWalletsRequest.CreateEmbeddedWalletsRequestFromJSONTyped;
exports.CreateEmbeddedWalletsRequestToJSON = CreateEmbeddedWalletsRequest.CreateEmbeddedWalletsRequestToJSON;
exports.CreateMfaTokenFromJSON = CreateMfaToken.CreateMfaTokenFromJSON;
exports.CreateMfaTokenFromJSONTyped = CreateMfaToken.CreateMfaTokenFromJSONTyped;
exports.CreateMfaTokenToJSON = CreateMfaToken.CreateMfaTokenToJSON;
exports.CreateTurnkeyEmbeddedWalletSpecificOptsFromJSON = CreateTurnkeyEmbeddedWalletSpecificOpts.CreateTurnkeyEmbeddedWalletSpecificOptsFromJSON;
exports.CreateTurnkeyEmbeddedWalletSpecificOptsFromJSONTyped = CreateTurnkeyEmbeddedWalletSpecificOpts.CreateTurnkeyEmbeddedWalletSpecificOptsFromJSONTyped;
exports.CreateTurnkeyEmbeddedWalletSpecificOptsToJSON = CreateTurnkeyEmbeddedWalletSpecificOpts.CreateTurnkeyEmbeddedWalletSpecificOptsToJSON;
exports.CreateUserEmbeddedWalletsFromFarcasterRequestFromJSON = CreateUserEmbeddedWalletsFromFarcasterRequest.CreateUserEmbeddedWalletsFromFarcasterRequestFromJSON;
exports.CreateUserEmbeddedWalletsFromFarcasterRequestFromJSONTyped = CreateUserEmbeddedWalletsFromFarcasterRequest.CreateUserEmbeddedWalletsFromFarcasterRequestFromJSONTyped;
exports.CreateUserEmbeddedWalletsFromFarcasterRequestToJSON = CreateUserEmbeddedWalletsFromFarcasterRequest.CreateUserEmbeddedWalletsFromFarcasterRequestToJSON;
exports.CreateUserEmbeddedWalletsRequestFromJSON = CreateUserEmbeddedWalletsRequest.CreateUserEmbeddedWalletsRequestFromJSON;
exports.CreateUserEmbeddedWalletsRequestFromJSONTyped = CreateUserEmbeddedWalletsRequest.CreateUserEmbeddedWalletsRequestFromJSONTyped;
exports.CreateUserEmbeddedWalletsRequestToJSON = CreateUserEmbeddedWalletsRequest.CreateUserEmbeddedWalletsRequestToJSON;
exports.CreateWalletAccountRequestFromJSON = CreateWalletAccountRequest.CreateWalletAccountRequestFromJSON;
exports.CreateWalletAccountRequestFromJSONTyped = CreateWalletAccountRequest.CreateWalletAccountRequestFromJSONTyped;
exports.CreateWalletAccountRequestToJSON = CreateWalletAccountRequest.CreateWalletAccountRequestToJSON;
exports.CurrencyFromJSON = Currency.CurrencyFromJSON;
exports.CurrencyFromJSONTyped = Currency.CurrencyFromJSONTyped;
exports.CurrencyToJSON = Currency.CurrencyToJSON;
Object.defineProperty(exports, 'CurrencyType', {
	enumerable: true,
	get: function () { return CurrencyType.CurrencyType; }
});
exports.CurrencyTypeFromJSON = CurrencyType.CurrencyTypeFromJSON;
exports.CurrencyTypeFromJSONTyped = CurrencyType.CurrencyTypeFromJSONTyped;
exports.CurrencyTypeToJSON = CurrencyType.CurrencyTypeToJSON;
Object.defineProperty(exports, 'CustomFieldType', {
	enumerable: true,
	get: function () { return CustomFieldType.CustomFieldType; }
});
exports.CustomFieldTypeFromJSON = CustomFieldType.CustomFieldTypeFromJSON;
exports.CustomFieldTypeFromJSONTyped = CustomFieldType.CustomFieldTypeFromJSONTyped;
exports.CustomFieldTypeToJSON = CustomFieldType.CustomFieldTypeToJSON;
exports.CustomFieldValidValueFromJSON = CustomFieldValidValue.CustomFieldValidValueFromJSON;
exports.CustomFieldValidValueFromJSONTyped = CustomFieldValidValue.CustomFieldValidValueFromJSONTyped;
exports.CustomFieldValidValueToJSON = CustomFieldValidValue.CustomFieldValidValueToJSON;
exports.CustomFieldValidationRulesFromJSON = CustomFieldValidationRules.CustomFieldValidationRulesFromJSON;
exports.CustomFieldValidationRulesFromJSONTyped = CustomFieldValidationRules.CustomFieldValidationRulesFromJSONTyped;
exports.CustomFieldValidationRulesToJSON = CustomFieldValidationRules.CustomFieldValidationRulesToJSON;
exports.DurationFromJSON = Duration.DurationFromJSON;
exports.DurationFromJSONTyped = Duration.DurationFromJSONTyped;
exports.DurationToJSON = Duration.DurationToJSON;
exports.DynamicJwtFromJSON = DynamicJwt.DynamicJwtFromJSON;
exports.DynamicJwtFromJSONTyped = DynamicJwt.DynamicJwtFromJSONTyped;
exports.DynamicJwtToJSON = DynamicJwt.DynamicJwtToJSON;
Object.defineProperty(exports, 'EcdsaValidatorOptions', {
	enumerable: true,
	get: function () { return EcdsaValidatorOptions.EcdsaValidatorOptions; }
});
exports.EcdsaValidatorOptionsFromJSON = EcdsaValidatorOptions.EcdsaValidatorOptionsFromJSON;
exports.EcdsaValidatorOptionsFromJSONTyped = EcdsaValidatorOptions.EcdsaValidatorOptionsFromJSONTyped;
exports.EcdsaValidatorOptionsToJSON = EcdsaValidatorOptions.EcdsaValidatorOptionsToJSON;
exports.EmailProviderResponseFromJSON = EmailProviderResponse.EmailProviderResponseFromJSON;
exports.EmailProviderResponseFromJSONTyped = EmailProviderResponse.EmailProviderResponseFromJSONTyped;
exports.EmailProviderResponseToJSON = EmailProviderResponse.EmailProviderResponseToJSON;
exports.EmailVerificationCreateRequestFromJSON = EmailVerificationCreateRequest.EmailVerificationCreateRequestFromJSON;
exports.EmailVerificationCreateRequestFromJSONTyped = EmailVerificationCreateRequest.EmailVerificationCreateRequestFromJSONTyped;
exports.EmailVerificationCreateRequestToJSON = EmailVerificationCreateRequest.EmailVerificationCreateRequestToJSON;
exports.EmailVerificationCreateResponseFromJSON = EmailVerificationCreateResponse.EmailVerificationCreateResponseFromJSON;
exports.EmailVerificationCreateResponseFromJSONTyped = EmailVerificationCreateResponse.EmailVerificationCreateResponseFromJSONTyped;
exports.EmailVerificationCreateResponseToJSON = EmailVerificationCreateResponse.EmailVerificationCreateResponseToJSON;
exports.EmailVerificationRetryRequestFromJSON = EmailVerificationRetryRequest.EmailVerificationRetryRequestFromJSON;
exports.EmailVerificationRetryRequestFromJSONTyped = EmailVerificationRetryRequest.EmailVerificationRetryRequestFromJSONTyped;
exports.EmailVerificationRetryRequestToJSON = EmailVerificationRetryRequest.EmailVerificationRetryRequestToJSON;
exports.EmailVerificationVerifyRequestFromJSON = EmailVerificationVerifyRequest.EmailVerificationVerifyRequestFromJSON;
exports.EmailVerificationVerifyRequestFromJSONTyped = EmailVerificationVerifyRequest.EmailVerificationVerifyRequestFromJSONTyped;
exports.EmailVerificationVerifyRequestToJSON = EmailVerificationVerifyRequest.EmailVerificationVerifyRequestToJSON;
exports.EmbeddedWalletAuthTokenFromJSON = EmbeddedWalletAuthToken.EmbeddedWalletAuthTokenFromJSON;
exports.EmbeddedWalletAuthTokenFromJSONTyped = EmbeddedWalletAuthToken.EmbeddedWalletAuthTokenFromJSONTyped;
exports.EmbeddedWalletAuthTokenToJSON = EmbeddedWalletAuthToken.EmbeddedWalletAuthTokenToJSON;
Object.defineProperty(exports, 'EmbeddedWalletAuthType', {
	enumerable: true,
	get: function () { return EmbeddedWalletAuthType.EmbeddedWalletAuthType; }
});
exports.EmbeddedWalletAuthTypeFromJSON = EmbeddedWalletAuthType.EmbeddedWalletAuthTypeFromJSON;
exports.EmbeddedWalletAuthTypeFromJSONTyped = EmbeddedWalletAuthType.EmbeddedWalletAuthTypeFromJSONTyped;
exports.EmbeddedWalletAuthTypeToJSON = EmbeddedWalletAuthType.EmbeddedWalletAuthTypeToJSON;
Object.defineProperty(exports, 'EmbeddedWalletChainEnum', {
	enumerable: true,
	get: function () { return EmbeddedWalletChainEnum.EmbeddedWalletChainEnum; }
});
exports.EmbeddedWalletChainEnumFromJSON = EmbeddedWalletChainEnum.EmbeddedWalletChainEnumFromJSON;
exports.EmbeddedWalletChainEnumFromJSONTyped = EmbeddedWalletChainEnum.EmbeddedWalletChainEnumFromJSONTyped;
exports.EmbeddedWalletChainEnumToJSON = EmbeddedWalletChainEnum.EmbeddedWalletChainEnumToJSON;
exports.EmbeddedWalletPasscodeClaimRequestFromJSON = EmbeddedWalletPasscodeClaimRequest.EmbeddedWalletPasscodeClaimRequestFromJSON;
exports.EmbeddedWalletPasscodeClaimRequestFromJSONTyped = EmbeddedWalletPasscodeClaimRequest.EmbeddedWalletPasscodeClaimRequestFromJSONTyped;
exports.EmbeddedWalletPasscodeClaimRequestToJSON = EmbeddedWalletPasscodeClaimRequest.EmbeddedWalletPasscodeClaimRequestToJSON;
Object.defineProperty(exports, 'EmbeddedWalletProviderEnum', {
	enumerable: true,
	get: function () { return EmbeddedWalletProviderEnum.EmbeddedWalletProviderEnum; }
});
exports.EmbeddedWalletProviderEnumFromJSON = EmbeddedWalletProviderEnum.EmbeddedWalletProviderEnumFromJSON;
exports.EmbeddedWalletProviderEnumFromJSONTyped = EmbeddedWalletProviderEnum.EmbeddedWalletProviderEnumFromJSONTyped;
exports.EmbeddedWalletProviderEnumToJSON = EmbeddedWalletProviderEnum.EmbeddedWalletProviderEnumToJSON;
exports.EmbeddedWalletSecretFromJSON = EmbeddedWalletSecret.EmbeddedWalletSecretFromJSON;
exports.EmbeddedWalletSecretFromJSONTyped = EmbeddedWalletSecret.EmbeddedWalletSecretFromJSONTyped;
exports.EmbeddedWalletSecretToJSON = EmbeddedWalletSecret.EmbeddedWalletSecretToJSON;
exports.EmbeddedWalletSecretWithUpdatedJwtFromJSON = EmbeddedWalletSecretWithUpdatedJwt.EmbeddedWalletSecretWithUpdatedJwtFromJSON;
exports.EmbeddedWalletSecretWithUpdatedJwtFromJSONTyped = EmbeddedWalletSecretWithUpdatedJwt.EmbeddedWalletSecretWithUpdatedJwtFromJSONTyped;
exports.EmbeddedWalletSecretWithUpdatedJwtToJSON = EmbeddedWalletSecretWithUpdatedJwt.EmbeddedWalletSecretWithUpdatedJwtToJSON;
exports.EmbeddedWalletSecretWithUpdatedJwtAllOfFromJSON = EmbeddedWalletSecretWithUpdatedJwtAllOf.EmbeddedWalletSecretWithUpdatedJwtAllOfFromJSON;
exports.EmbeddedWalletSecretWithUpdatedJwtAllOfFromJSONTyped = EmbeddedWalletSecretWithUpdatedJwtAllOf.EmbeddedWalletSecretWithUpdatedJwtAllOfFromJSONTyped;
exports.EmbeddedWalletSecretWithUpdatedJwtAllOfToJSON = EmbeddedWalletSecretWithUpdatedJwtAllOf.EmbeddedWalletSecretWithUpdatedJwtAllOfToJSON;
Object.defineProperty(exports, 'EmbeddedWalletSecurityMethod', {
	enumerable: true,
	get: function () { return EmbeddedWalletSecurityMethod.EmbeddedWalletSecurityMethod; }
});
exports.EmbeddedWalletSecurityMethodFromJSON = EmbeddedWalletSecurityMethod.EmbeddedWalletSecurityMethodFromJSON;
exports.EmbeddedWalletSecurityMethodFromJSONTyped = EmbeddedWalletSecurityMethod.EmbeddedWalletSecurityMethodFromJSONTyped;
exports.EmbeddedWalletSecurityMethodToJSON = EmbeddedWalletSecurityMethod.EmbeddedWalletSecurityMethodToJSON;
Object.defineProperty(exports, 'EmbeddedWalletVersionEnum', {
	enumerable: true,
	get: function () { return EmbeddedWalletVersionEnum.EmbeddedWalletVersionEnum; }
});
exports.EmbeddedWalletVersionEnumFromJSON = EmbeddedWalletVersionEnum.EmbeddedWalletVersionEnumFromJSON;
exports.EmbeddedWalletVersionEnumFromJSONTyped = EmbeddedWalletVersionEnum.EmbeddedWalletVersionEnumFromJSONTyped;
exports.EmbeddedWalletVersionEnumToJSON = EmbeddedWalletVersionEnum.EmbeddedWalletVersionEnumToJSON;
Object.defineProperty(exports, 'EnvironmentEnum', {
	enumerable: true,
	get: function () { return EnvironmentEnum.EnvironmentEnum; }
});
exports.EnvironmentEnumFromJSON = EnvironmentEnum.EnvironmentEnumFromJSON;
exports.EnvironmentEnumFromJSONTyped = EnvironmentEnum.EnvironmentEnumFromJSONTyped;
exports.EnvironmentEnumToJSON = EnvironmentEnum.EnvironmentEnumToJSON;
exports.ErrorMessageWithCodeFromJSON = ErrorMessageWithCode.ErrorMessageWithCodeFromJSON;
exports.ErrorMessageWithCodeFromJSONTyped = ErrorMessageWithCode.ErrorMessageWithCodeFromJSONTyped;
exports.ErrorMessageWithCodeToJSON = ErrorMessageWithCode.ErrorMessageWithCodeToJSON;
exports.ExportEmbeddedWalletResponseFromJSON = ExportEmbeddedWalletResponse.ExportEmbeddedWalletResponseFromJSON;
exports.ExportEmbeddedWalletResponseFromJSONTyped = ExportEmbeddedWalletResponse.ExportEmbeddedWalletResponseFromJSONTyped;
exports.ExportEmbeddedWalletResponseToJSON = ExportEmbeddedWalletResponse.ExportEmbeddedWalletResponseToJSON;
exports.ExternalAuthFromJSON = ExternalAuth.ExternalAuthFromJSON;
exports.ExternalAuthFromJSONTyped = ExternalAuth.ExternalAuthFromJSONTyped;
exports.ExternalAuthToJSON = ExternalAuth.ExternalAuthToJSON;
exports.ExternalAuthSigninRequestFromJSON = ExternalAuthSigninRequest.ExternalAuthSigninRequestFromJSON;
exports.ExternalAuthSigninRequestFromJSONTyped = ExternalAuthSigninRequest.ExternalAuthSigninRequestFromJSONTyped;
exports.ExternalAuthSigninRequestToJSON = ExternalAuthSigninRequest.ExternalAuthSigninRequestToJSON;
exports.FarcasterSignInRequestFromJSON = FarcasterSignInRequest.FarcasterSignInRequestFromJSON;
exports.FarcasterSignInRequestFromJSONTyped = FarcasterSignInRequest.FarcasterSignInRequestFromJSONTyped;
exports.FarcasterSignInRequestToJSON = FarcasterSignInRequest.FarcasterSignInRequestToJSON;
exports.ForbiddenFromJSON = Forbidden.ForbiddenFromJSON;
exports.ForbiddenFromJSONTyped = Forbidden.ForbiddenFromJSONTyped;
exports.ForbiddenToJSON = Forbidden.ForbiddenToJSON;
exports.ForbiddenErrorPayloadFromJSON = ForbiddenErrorPayload.ForbiddenErrorPayloadFromJSON;
exports.ForbiddenErrorPayloadFromJSONTyped = ForbiddenErrorPayload.ForbiddenErrorPayloadFromJSONTyped;
exports.ForbiddenErrorPayloadToJSON = ForbiddenErrorPayload.ForbiddenErrorPayloadToJSON;
exports.ForbiddenWithErrorAndPayloadFromJSON = ForbiddenWithErrorAndPayload.ForbiddenWithErrorAndPayloadFromJSON;
exports.ForbiddenWithErrorAndPayloadFromJSONTyped = ForbiddenWithErrorAndPayload.ForbiddenWithErrorAndPayloadFromJSONTyped;
exports.ForbiddenWithErrorAndPayloadToJSON = ForbiddenWithErrorAndPayload.ForbiddenWithErrorAndPayloadToJSON;
exports.FrameworkSettingsFromJSON = FrameworkSettings.FrameworkSettingsFromJSON;
exports.FrameworkSettingsFromJSONTyped = FrameworkSettings.FrameworkSettingsFromJSONTyped;
exports.FrameworkSettingsToJSON = FrameworkSettings.FrameworkSettingsToJSON;
exports.GetUserPasskeysResponseFromJSON = GetUserPasskeysResponse.GetUserPasskeysResponseFromJSON;
exports.GetUserPasskeysResponseFromJSONTyped = GetUserPasskeysResponse.GetUserPasskeysResponseFromJSONTyped;
exports.GetUserPasskeysResponseToJSON = GetUserPasskeysResponse.GetUserPasskeysResponseToJSON;
exports.HCaptchaSettingsFromJSON = HCaptchaSettings.HCaptchaSettingsFromJSON;
exports.HCaptchaSettingsFromJSONTyped = HCaptchaSettings.HCaptchaSettingsFromJSONTyped;
exports.HCaptchaSettingsToJSON = HCaptchaSettings.HCaptchaSettingsToJSON;
Object.defineProperty(exports, 'HardwareWalletEnum', {
	enumerable: true,
	get: function () { return HardwareWalletEnum.HardwareWalletEnum; }
});
exports.HardwareWalletEnumFromJSON = HardwareWalletEnum.HardwareWalletEnumFromJSON;
exports.HardwareWalletEnumFromJSONTyped = HardwareWalletEnum.HardwareWalletEnumFromJSONTyped;
exports.HardwareWalletEnumToJSON = HardwareWalletEnum.HardwareWalletEnumToJSON;
exports.HardwareWalletPropertiesFromJSON = HardwareWalletProperties.HardwareWalletPropertiesFromJSON;
exports.HardwareWalletPropertiesFromJSONTyped = HardwareWalletProperties.HardwareWalletPropertiesFromJSONTyped;
exports.HardwareWalletPropertiesToJSON = HardwareWalletProperties.HardwareWalletPropertiesToJSON;
exports.HealthcheckResponseFromJSON = HealthcheckResponse.HealthcheckResponseFromJSON;
exports.HealthcheckResponseFromJSONTyped = HealthcheckResponse.HealthcheckResponseFromJSONTyped;
exports.HealthcheckResponseToJSON = HealthcheckResponse.HealthcheckResponseToJSON;
Object.defineProperty(exports, 'HealthcheckStatus', {
	enumerable: true,
	get: function () { return HealthcheckStatus.HealthcheckStatus; }
});
exports.HealthcheckStatusFromJSON = HealthcheckStatus.HealthcheckStatusFromJSON;
exports.HealthcheckStatusFromJSONTyped = HealthcheckStatus.HealthcheckStatusFromJSONTyped;
exports.HealthcheckStatusToJSON = HealthcheckStatus.HealthcheckStatusToJSON;
exports.InitEmailAuthRequestFromJSON = InitEmailAuthRequest.InitEmailAuthRequestFromJSON;
exports.InitEmailAuthRequestFromJSONTyped = InitEmailAuthRequest.InitEmailAuthRequestFromJSONTyped;
exports.InitEmailAuthRequestToJSON = InitEmailAuthRequest.InitEmailAuthRequestToJSON;
exports.InitEmailAuthResponseFromJSON = InitEmailAuthResponse.InitEmailAuthResponseFromJSON;
exports.InitEmailAuthResponseFromJSONTyped = InitEmailAuthResponse.InitEmailAuthResponseFromJSONTyped;
exports.InitEmailAuthResponseToJSON = InitEmailAuthResponse.InitEmailAuthResponseToJSON;
exports.InitPasskeyRecoveryRequestFromJSON = InitPasskeyRecoveryRequest.InitPasskeyRecoveryRequestFromJSON;
exports.InitPasskeyRecoveryRequestFromJSONTyped = InitPasskeyRecoveryRequest.InitPasskeyRecoveryRequestFromJSONTyped;
exports.InitPasskeyRecoveryRequestToJSON = InitPasskeyRecoveryRequest.InitPasskeyRecoveryRequestToJSON;
exports.InitPasskeyRecoveryResponseFromJSON = InitPasskeyRecoveryResponse.InitPasskeyRecoveryResponseFromJSON;
exports.InitPasskeyRecoveryResponseFromJSONTyped = InitPasskeyRecoveryResponse.InitPasskeyRecoveryResponseFromJSONTyped;
exports.InitPasskeyRecoveryResponseToJSON = InitPasskeyRecoveryResponse.InitPasskeyRecoveryResponseToJSON;
exports.IntegrationSettingFromJSON = IntegrationSetting.IntegrationSettingFromJSON;
exports.IntegrationSettingFromJSONTyped = IntegrationSetting.IntegrationSettingFromJSONTyped;
exports.IntegrationSettingToJSON = IntegrationSetting.IntegrationSettingToJSON;
exports.InternalServerErrorFromJSON = InternalServerError.InternalServerErrorFromJSON;
exports.InternalServerErrorFromJSONTyped = InternalServerError.InternalServerErrorFromJSONTyped;
exports.InternalServerErrorToJSON = InternalServerError.InternalServerErrorToJSON;
exports.JwksKeyFromJSON = JwksKey.JwksKeyFromJSON;
exports.JwksKeyFromJSONTyped = JwksKey.JwksKeyFromJSONTyped;
exports.JwksKeyToJSON = JwksKey.JwksKeyToJSON;
exports.JwksResponseFromJSON = JwksResponse.JwksResponseFromJSON;
exports.JwksResponseFromJSONTyped = JwksResponse.JwksResponseFromJSONTyped;
exports.JwksResponseToJSON = JwksResponse.JwksResponseToJSON;
exports.JwtBlockchainAccountFromJSON = JwtBlockchainAccount.JwtBlockchainAccountFromJSON;
exports.JwtBlockchainAccountFromJSONTyped = JwtBlockchainAccount.JwtBlockchainAccountFromJSONTyped;
exports.JwtBlockchainAccountToJSON = JwtBlockchainAccount.JwtBlockchainAccountToJSON;
exports.JwtPayloadDeprecatedInfoFromJSON = JwtPayloadDeprecatedInfo.JwtPayloadDeprecatedInfoFromJSON;
exports.JwtPayloadDeprecatedInfoFromJSONTyped = JwtPayloadDeprecatedInfo.JwtPayloadDeprecatedInfoFromJSONTyped;
exports.JwtPayloadDeprecatedInfoToJSON = JwtPayloadDeprecatedInfo.JwtPayloadDeprecatedInfoToJSON;
exports.JwtVerifiedCredentialFromJSON = JwtVerifiedCredential.JwtVerifiedCredentialFromJSON;
exports.JwtVerifiedCredentialFromJSONTyped = JwtVerifiedCredential.JwtVerifiedCredentialFromJSONTyped;
exports.JwtVerifiedCredentialToJSON = JwtVerifiedCredential.JwtVerifiedCredentialToJSON;
Object.defineProperty(exports, 'JwtVerifiedCredentialFormatEnum', {
	enumerable: true,
	get: function () { return JwtVerifiedCredentialFormatEnum.JwtVerifiedCredentialFormatEnum; }
});
exports.JwtVerifiedCredentialFormatEnumFromJSON = JwtVerifiedCredentialFormatEnum.JwtVerifiedCredentialFormatEnumFromJSON;
exports.JwtVerifiedCredentialFormatEnumFromJSONTyped = JwtVerifiedCredentialFormatEnum.JwtVerifiedCredentialFormatEnumFromJSONTyped;
exports.JwtVerifiedCredentialFormatEnumToJSON = JwtVerifiedCredentialFormatEnum.JwtVerifiedCredentialFormatEnumToJSON;
Object.defineProperty(exports, 'KycFieldType', {
	enumerable: true,
	get: function () { return KycFieldType.KycFieldType; }
});
exports.KycFieldTypeFromJSON = KycFieldType.KycFieldTypeFromJSON;
exports.KycFieldTypeFromJSONTyped = KycFieldType.KycFieldTypeFromJSONTyped;
exports.KycFieldTypeToJSON = KycFieldType.KycFieldTypeToJSON;
exports.MFAAuthPasskeyDeviceGetResponseFromJSON = MFAAuthPasskeyDeviceGetResponse.MFAAuthPasskeyDeviceGetResponseFromJSON;
exports.MFAAuthPasskeyDeviceGetResponseFromJSONTyped = MFAAuthPasskeyDeviceGetResponse.MFAAuthPasskeyDeviceGetResponseFromJSONTyped;
exports.MFAAuthPasskeyDeviceGetResponseToJSON = MFAAuthPasskeyDeviceGetResponse.MFAAuthPasskeyDeviceGetResponseToJSON;
exports.MFAAuthPasskeyDeviceGetResponseAssertionFromJSON = MFAAuthPasskeyDeviceGetResponseAssertion.MFAAuthPasskeyDeviceGetResponseAssertionFromJSON;
exports.MFAAuthPasskeyDeviceGetResponseAssertionFromJSONTyped = MFAAuthPasskeyDeviceGetResponseAssertion.MFAAuthPasskeyDeviceGetResponseAssertionFromJSONTyped;
exports.MFAAuthPasskeyDeviceGetResponseAssertionToJSON = MFAAuthPasskeyDeviceGetResponseAssertion.MFAAuthPasskeyDeviceGetResponseAssertionToJSON;
exports.MFAAuthPasskeyDevicePostRequestFromJSON = MFAAuthPasskeyDevicePostRequest.MFAAuthPasskeyDevicePostRequestFromJSON;
exports.MFAAuthPasskeyDevicePostRequestFromJSONTyped = MFAAuthPasskeyDevicePostRequest.MFAAuthPasskeyDevicePostRequestFromJSONTyped;
exports.MFAAuthPasskeyDevicePostRequestToJSON = MFAAuthPasskeyDevicePostRequest.MFAAuthPasskeyDevicePostRequestToJSON;
exports.MFAAuthRecoveryDevicePostRequestFromJSON = MFAAuthRecoveryDevicePostRequest.MFAAuthRecoveryDevicePostRequestFromJSON;
exports.MFAAuthRecoveryDevicePostRequestFromJSONTyped = MFAAuthRecoveryDevicePostRequest.MFAAuthRecoveryDevicePostRequestFromJSONTyped;
exports.MFAAuthRecoveryDevicePostRequestToJSON = MFAAuthRecoveryDevicePostRequest.MFAAuthRecoveryDevicePostRequestToJSON;
exports.MFAAuthTotpDevicePostRequestFromJSON = MFAAuthTotpDevicePostRequest.MFAAuthTotpDevicePostRequestFromJSON;
exports.MFAAuthTotpDevicePostRequestFromJSONTyped = MFAAuthTotpDevicePostRequest.MFAAuthTotpDevicePostRequestFromJSONTyped;
exports.MFAAuthTotpDevicePostRequestToJSON = MFAAuthTotpDevicePostRequest.MFAAuthTotpDevicePostRequestToJSON;
exports.MFADeviceFromJSON = MFADevice.MFADeviceFromJSON;
exports.MFADeviceFromJSONTyped = MFADevice.MFADeviceFromJSONTyped;
exports.MFADeviceToJSON = MFADevice.MFADeviceToJSON;
Object.defineProperty(exports, 'MFADeviceType', {
	enumerable: true,
	get: function () { return MFADeviceType.MFADeviceType; }
});
exports.MFADeviceTypeFromJSON = MFADeviceType.MFADeviceTypeFromJSON;
exports.MFADeviceTypeFromJSONTyped = MFADeviceType.MFADeviceTypeFromJSONTyped;
exports.MFADeviceTypeToJSON = MFADeviceType.MFADeviceTypeToJSON;
exports.MFAGetRecoveryCodesResponseFromJSON = MFAGetRecoveryCodesResponse.MFAGetRecoveryCodesResponseFromJSON;
exports.MFAGetRecoveryCodesResponseFromJSONTyped = MFAGetRecoveryCodesResponse.MFAGetRecoveryCodesResponseFromJSONTyped;
exports.MFAGetRecoveryCodesResponseToJSON = MFAGetRecoveryCodesResponse.MFAGetRecoveryCodesResponseToJSON;
exports.MFAListDevicesResponseFromJSON = MFAListDevicesResponse.MFAListDevicesResponseFromJSON;
exports.MFAListDevicesResponseFromJSONTyped = MFAListDevicesResponse.MFAListDevicesResponseFromJSONTyped;
exports.MFAListDevicesResponseToJSON = MFAListDevicesResponse.MFAListDevicesResponseToJSON;
exports.MFARegenRecoveryCodesResponseFromJSON = MFARegenRecoveryCodesResponse.MFARegenRecoveryCodesResponseFromJSON;
exports.MFARegenRecoveryCodesResponseFromJSONTyped = MFARegenRecoveryCodesResponse.MFARegenRecoveryCodesResponseFromJSONTyped;
exports.MFARegenRecoveryCodesResponseToJSON = MFARegenRecoveryCodesResponse.MFARegenRecoveryCodesResponseToJSON;
exports.MFARegisterPasskeyDeviceGetResponseFromJSON = MFARegisterPasskeyDeviceGetResponse.MFARegisterPasskeyDeviceGetResponseFromJSON;
exports.MFARegisterPasskeyDeviceGetResponseFromJSONTyped = MFARegisterPasskeyDeviceGetResponse.MFARegisterPasskeyDeviceGetResponseFromJSONTyped;
exports.MFARegisterPasskeyDeviceGetResponseToJSON = MFARegisterPasskeyDeviceGetResponse.MFARegisterPasskeyDeviceGetResponseToJSON;
exports.MFARegisterPasskeyDeviceGetResponseRegistrationFromJSON = MFARegisterPasskeyDeviceGetResponseRegistration.MFARegisterPasskeyDeviceGetResponseRegistrationFromJSON;
exports.MFARegisterPasskeyDeviceGetResponseRegistrationFromJSONTyped = MFARegisterPasskeyDeviceGetResponseRegistration.MFARegisterPasskeyDeviceGetResponseRegistrationFromJSONTyped;
exports.MFARegisterPasskeyDeviceGetResponseRegistrationToJSON = MFARegisterPasskeyDeviceGetResponseRegistration.MFARegisterPasskeyDeviceGetResponseRegistrationToJSON;
exports.MFARegisterPasskeyDevicePostRequestFromJSON = MFARegisterPasskeyDevicePostRequest.MFARegisterPasskeyDevicePostRequestFromJSON;
exports.MFARegisterPasskeyDevicePostRequestFromJSONTyped = MFARegisterPasskeyDevicePostRequest.MFARegisterPasskeyDevicePostRequestFromJSONTyped;
exports.MFARegisterPasskeyDevicePostRequestToJSON = MFARegisterPasskeyDevicePostRequest.MFARegisterPasskeyDevicePostRequestToJSON;
exports.MFARegisterTotpDeviceGetResponseFromJSON = MFARegisterTotpDeviceGetResponse.MFARegisterTotpDeviceGetResponseFromJSON;
exports.MFARegisterTotpDeviceGetResponseFromJSONTyped = MFARegisterTotpDeviceGetResponse.MFARegisterTotpDeviceGetResponseFromJSONTyped;
exports.MFARegisterTotpDeviceGetResponseToJSON = MFARegisterTotpDeviceGetResponse.MFARegisterTotpDeviceGetResponseToJSON;
exports.MFARegisterTotpDevicePostRequestFromJSON = MFARegisterTotpDevicePostRequest.MFARegisterTotpDevicePostRequestFromJSON;
exports.MFARegisterTotpDevicePostRequestFromJSONTyped = MFARegisterTotpDevicePostRequest.MFARegisterTotpDevicePostRequestFromJSONTyped;
exports.MFARegisterTotpDevicePostRequestToJSON = MFARegisterTotpDevicePostRequest.MFARegisterTotpDevicePostRequestToJSON;
exports.MFASettingsFromJSON = MFASettings.MFASettingsFromJSON;
exports.MFASettingsFromJSONTyped = MFASettings.MFASettingsFromJSONTyped;
exports.MFASettingsToJSON = MFASettings.MFASettingsToJSON;
exports.MFAUpdateDeviceRequestFromJSON = MFAUpdateDeviceRequest.MFAUpdateDeviceRequestFromJSON;
exports.MFAUpdateDeviceRequestFromJSONTyped = MFAUpdateDeviceRequest.MFAUpdateDeviceRequestFromJSONTyped;
exports.MFAUpdateDeviceRequestToJSON = MFAUpdateDeviceRequest.MFAUpdateDeviceRequestToJSON;
exports.MergeConflictsFromJSON = MergeConflicts.MergeConflictsFromJSON;
exports.MergeConflictsFromJSONTyped = MergeConflicts.MergeConflictsFromJSONTyped;
exports.MergeConflictsToJSON = MergeConflicts.MergeConflictsToJSON;
exports.MergeUserFromJSON = MergeUser.MergeUserFromJSON;
exports.MergeUserFromJSONTyped = MergeUser.MergeUserFromJSONTyped;
exports.MergeUserToJSON = MergeUser.MergeUserToJSON;
exports.MergeUserConflictFromJSON = MergeUserConflict.MergeUserConflictFromJSON;
exports.MergeUserConflictFromJSONTyped = MergeUserConflict.MergeUserConflictFromJSONTyped;
exports.MergeUserConflictToJSON = MergeUserConflict.MergeUserConflictToJSON;
exports.MergeUserConflictResolutionFromJSON = MergeUserConflictResolution.MergeUserConflictResolutionFromJSON;
exports.MergeUserConflictResolutionFromJSONTyped = MergeUserConflictResolution.MergeUserConflictResolutionFromJSONTyped;
exports.MergeUserConflictResolutionToJSON = MergeUserConflictResolution.MergeUserConflictResolutionToJSON;
exports.MergeUserConflictResolutionsFromJSON = MergeUserConflictResolutions.MergeUserConflictResolutionsFromJSON;
exports.MergeUserConflictResolutionsFromJSONTyped = MergeUserConflictResolutions.MergeUserConflictResolutionsFromJSONTyped;
exports.MergeUserConflictResolutionsToJSON = MergeUserConflictResolutions.MergeUserConflictResolutionsToJSON;
exports.MethodNotAllowedFromJSON = MethodNotAllowed.MethodNotAllowedFromJSON;
exports.MethodNotAllowedFromJSONTyped = MethodNotAllowed.MethodNotAllowedFromJSONTyped;
exports.MethodNotAllowedToJSON = MethodNotAllowed.MethodNotAllowedToJSON;
Object.defineProperty(exports, 'MfaBackupCodeAcknowledgement', {
	enumerable: true,
	get: function () { return MfaBackupCodeAcknowledgement.MfaBackupCodeAcknowledgement; }
});
exports.MfaBackupCodeAcknowledgementFromJSON = MfaBackupCodeAcknowledgement.MfaBackupCodeAcknowledgementFromJSON;
exports.MfaBackupCodeAcknowledgementFromJSONTyped = MfaBackupCodeAcknowledgement.MfaBackupCodeAcknowledgementFromJSONTyped;
exports.MfaBackupCodeAcknowledgementToJSON = MfaBackupCodeAcknowledgement.MfaBackupCodeAcknowledgementToJSON;
exports.MinifiedDynamicJwtFromJSON = MinifiedDynamicJwt.MinifiedDynamicJwtFromJSON;
exports.MinifiedDynamicJwtFromJSONTyped = MinifiedDynamicJwt.MinifiedDynamicJwtFromJSONTyped;
exports.MinifiedDynamicJwtToJSON = MinifiedDynamicJwt.MinifiedDynamicJwtToJSON;
exports.MobileSettingsFromJSON = MobileSettings.MobileSettingsFromJSON;
exports.MobileSettingsFromJSONTyped = MobileSettings.MobileSettingsFromJSONTyped;
exports.MobileSettingsToJSON = MobileSettings.MobileSettingsToJSON;
exports.NameServiceFromJSON = NameService.NameServiceFromJSON;
exports.NameServiceFromJSONTyped = NameService.NameServiceFromJSONTyped;
exports.NameServiceToJSON = NameService.NameServiceToJSON;
exports.NameServiceDataFromJSON = NameServiceData.NameServiceDataFromJSON;
exports.NameServiceDataFromJSONTyped = NameServiceData.NameServiceDataFromJSONTyped;
exports.NameServiceDataToJSON = NameServiceData.NameServiceDataToJSON;
exports.NativeCurrencyFromJSON = NativeCurrency.NativeCurrencyFromJSON;
exports.NativeCurrencyFromJSONTyped = NativeCurrency.NativeCurrencyFromJSONTyped;
exports.NativeCurrencyToJSON = NativeCurrency.NativeCurrencyToJSON;
exports.NetworkFromJSON = Network.NetworkFromJSON;
exports.NetworkFromJSONTyped = Network.NetworkFromJSONTyped;
exports.NetworkToJSON = Network.NetworkToJSON;
exports.NetworkConfigurationFromJSON = NetworkConfiguration.NetworkConfigurationFromJSON;
exports.NetworkConfigurationFromJSONTyped = NetworkConfiguration.NetworkConfigurationFromJSONTyped;
exports.NetworkConfigurationToJSON = NetworkConfiguration.NetworkConfigurationToJSON;
exports.NetworkConfigurationResponseFromJSON = NetworkConfigurationResponse.NetworkConfigurationResponseFromJSON;
exports.NetworkConfigurationResponseFromJSONTyped = NetworkConfigurationResponse.NetworkConfigurationResponseFromJSONTyped;
exports.NetworkConfigurationResponseToJSON = NetworkConfigurationResponse.NetworkConfigurationResponseToJSON;
exports.NextJsSettingsFromJSON = NextJsSettings.NextJsSettingsFromJSON;
exports.NextJsSettingsFromJSONTyped = NextJsSettings.NextJsSettingsFromJSONTyped;
exports.NextJsSettingsToJSON = NextJsSettings.NextJsSettingsToJSON;
Object.defineProperty(exports, 'NextViewEnum', {
	enumerable: true,
	get: function () { return NextViewEnum.NextViewEnum; }
});
exports.NextViewEnumFromJSON = NextViewEnum.NextViewEnumFromJSON;
exports.NextViewEnumFromJSONTyped = NextViewEnum.NextViewEnumFromJSONTyped;
exports.NextViewEnumToJSON = NextViewEnum.NextViewEnumToJSON;
exports.NonceResponseFromJSON = NonceResponse.NonceResponseFromJSON;
exports.NonceResponseFromJSONTyped = NonceResponse.NonceResponseFromJSONTyped;
exports.NonceResponseToJSON = NonceResponse.NonceResponseToJSON;
exports.NotFoundFromJSON = NotFound.NotFoundFromJSON;
exports.NotFoundFromJSONTyped = NotFound.NotFoundFromJSONTyped;
exports.NotFoundToJSON = NotFound.NotFoundToJSON;
exports.OAuthAccountFromJSON = OAuthAccount.OAuthAccountFromJSON;
exports.OAuthAccountFromJSONTyped = OAuthAccount.OAuthAccountFromJSONTyped;
exports.OAuthAccountToJSON = OAuthAccount.OAuthAccountToJSON;
exports.OauthInitAuthRequestFromJSON = OauthInitAuthRequest.OauthInitAuthRequestFromJSON;
exports.OauthInitAuthRequestFromJSONTyped = OauthInitAuthRequest.OauthInitAuthRequestFromJSONTyped;
exports.OauthInitAuthRequestToJSON = OauthInitAuthRequest.OauthInitAuthRequestToJSON;
exports.OauthProviderLoginUrlFromJSON = OauthProviderLoginUrl.OauthProviderLoginUrlFromJSON;
exports.OauthProviderLoginUrlFromJSONTyped = OauthProviderLoginUrl.OauthProviderLoginUrlFromJSONTyped;
exports.OauthProviderLoginUrlToJSON = OauthProviderLoginUrl.OauthProviderLoginUrlToJSON;
exports.OauthProviderRequestFromJSON = OauthProviderRequest.OauthProviderRequestFromJSON;
exports.OauthProviderRequestFromJSONTyped = OauthProviderRequest.OauthProviderRequestFromJSONTyped;
exports.OauthProviderRequestToJSON = OauthProviderRequest.OauthProviderRequestToJSON;
exports.OauthRequestFromJSON = OauthRequest.OauthRequestFromJSON;
exports.OauthRequestFromJSONTyped = OauthRequest.OauthRequestFromJSONTyped;
exports.OauthRequestToJSON = OauthRequest.OauthRequestToJSON;
exports.OauthResultRequestFromJSON = OauthResultRequest.OauthResultRequestFromJSON;
exports.OauthResultRequestFromJSONTyped = OauthResultRequest.OauthResultRequestFromJSONTyped;
exports.OauthResultRequestToJSON = OauthResultRequest.OauthResultRequestToJSON;
exports.OauthResultResponseFromJSON = OauthResultResponse.OauthResultResponseFromJSON;
exports.OauthResultResponseFromJSONTyped = OauthResultResponse.OauthResultResponseFromJSONTyped;
exports.OauthResultResponseToJSON = OauthResultResponse.OauthResultResponseToJSON;
Object.defineProperty(exports, 'OauthResultStatus', {
	enumerable: true,
	get: function () { return OauthResultStatus.OauthResultStatus; }
});
exports.OauthResultStatusFromJSON = OauthResultStatus.OauthResultStatusFromJSON;
exports.OauthResultStatusFromJSONTyped = OauthResultStatus.OauthResultStatusFromJSONTyped;
exports.OauthResultStatusToJSON = OauthResultStatus.OauthResultStatusToJSON;
exports.OnrampConfigurationFromJSON = OnrampConfiguration.OnrampConfigurationFromJSON;
exports.OnrampConfigurationFromJSONTyped = OnrampConfiguration.OnrampConfigurationFromJSONTyped;
exports.OnrampConfigurationToJSON = OnrampConfiguration.OnrampConfigurationToJSON;
exports.PasskeyAuthRequestFromJSON = PasskeyAuthRequest.PasskeyAuthRequestFromJSON;
exports.PasskeyAuthRequestFromJSONTyped = PasskeyAuthRequest.PasskeyAuthRequestFromJSONTyped;
exports.PasskeyAuthRequestToJSON = PasskeyAuthRequest.PasskeyAuthRequestToJSON;
exports.PasskeyRegisterVerifyRequestFromJSON = PasskeyRegisterVerifyRequest.PasskeyRegisterVerifyRequestFromJSON;
exports.PasskeyRegisterVerifyRequestFromJSONTyped = PasskeyRegisterVerifyRequest.PasskeyRegisterVerifyRequestFromJSONTyped;
exports.PasskeyRegisterVerifyRequestToJSON = PasskeyRegisterVerifyRequest.PasskeyRegisterVerifyRequestToJSON;
exports.PasskeyRegistrationCredentialFromJSON = PasskeyRegistrationCredential.PasskeyRegistrationCredentialFromJSON;
exports.PasskeyRegistrationCredentialFromJSONTyped = PasskeyRegistrationCredential.PasskeyRegistrationCredentialFromJSONTyped;
exports.PasskeyRegistrationCredentialToJSON = PasskeyRegistrationCredential.PasskeyRegistrationCredentialToJSON;
exports.PasskeyStorageFromJSON = PasskeyStorage.PasskeyStorageFromJSON;
exports.PasskeyStorageFromJSONTyped = PasskeyStorage.PasskeyStorageFromJSONTyped;
exports.PasskeyStorageToJSON = PasskeyStorage.PasskeyStorageToJSON;
Object.defineProperty(exports, 'PasswordSourceTypeEnum', {
	enumerable: true,
	get: function () { return PasswordSourceTypeEnum.PasswordSourceTypeEnum; }
});
exports.PasswordSourceTypeEnumFromJSON = PasswordSourceTypeEnum.PasswordSourceTypeEnumFromJSON;
exports.PasswordSourceTypeEnumFromJSONTyped = PasswordSourceTypeEnum.PasswordSourceTypeEnumFromJSONTyped;
exports.PasswordSourceTypeEnumToJSON = PasswordSourceTypeEnum.PasswordSourceTypeEnumToJSON;
exports.PrefetchRequestFromJSON = PrefetchRequest.PrefetchRequestFromJSON;
exports.PrefetchRequestFromJSONTyped = PrefetchRequest.PrefetchRequestFromJSONTyped;
exports.PrefetchRequestToJSON = PrefetchRequest.PrefetchRequestToJSON;
exports.ProjectSettingsFromJSON = ProjectSettings.ProjectSettingsFromJSON;
exports.ProjectSettingsFromJSONTyped = ProjectSettings.ProjectSettingsFromJSONTyped;
exports.ProjectSettingsToJSON = ProjectSettings.ProjectSettingsToJSON;
exports.ProjectSettingsChainsFromJSON = ProjectSettingsChains.ProjectSettingsChainsFromJSON;
exports.ProjectSettingsChainsFromJSONTyped = ProjectSettingsChains.ProjectSettingsChainsFromJSONTyped;
exports.ProjectSettingsChainsToJSON = ProjectSettingsChains.ProjectSettingsChainsToJSON;
exports.ProjectSettingsDesignFromJSON = ProjectSettingsDesign.ProjectSettingsDesignFromJSON;
exports.ProjectSettingsDesignFromJSONTyped = ProjectSettingsDesign.ProjectSettingsDesignFromJSONTyped;
exports.ProjectSettingsDesignToJSON = ProjectSettingsDesign.ProjectSettingsDesignToJSON;
exports.ProjectSettingsDesignButtonFromJSON = ProjectSettingsDesignButton.ProjectSettingsDesignButtonFromJSON;
exports.ProjectSettingsDesignButtonFromJSONTyped = ProjectSettingsDesignButton.ProjectSettingsDesignButtonFromJSONTyped;
exports.ProjectSettingsDesignButtonToJSON = ProjectSettingsDesignButton.ProjectSettingsDesignButtonToJSON;
exports.ProjectSettingsDesignModalFromJSON = ProjectSettingsDesignModal.ProjectSettingsDesignModalFromJSON;
exports.ProjectSettingsDesignModalFromJSONTyped = ProjectSettingsDesignModal.ProjectSettingsDesignModalFromJSONTyped;
exports.ProjectSettingsDesignModalToJSON = ProjectSettingsDesignModal.ProjectSettingsDesignModalToJSON;
exports.ProjectSettingsDesignWidgetFromJSON = ProjectSettingsDesignWidget.ProjectSettingsDesignWidgetFromJSON;
exports.ProjectSettingsDesignWidgetFromJSONTyped = ProjectSettingsDesignWidget.ProjectSettingsDesignWidgetFromJSONTyped;
exports.ProjectSettingsDesignWidgetToJSON = ProjectSettingsDesignWidget.ProjectSettingsDesignWidgetToJSON;
exports.ProjectSettingsGeneralFromJSON = ProjectSettingsGeneral.ProjectSettingsGeneralFromJSON;
exports.ProjectSettingsGeneralFromJSONTyped = ProjectSettingsGeneral.ProjectSettingsGeneralFromJSONTyped;
exports.ProjectSettingsGeneralToJSON = ProjectSettingsGeneral.ProjectSettingsGeneralToJSON;
exports.ProjectSettingsKycFromJSON = ProjectSettingsKyc.ProjectSettingsKycFromJSON;
exports.ProjectSettingsKycFromJSONTyped = ProjectSettingsKyc.ProjectSettingsKycFromJSONTyped;
exports.ProjectSettingsKycToJSON = ProjectSettingsKyc.ProjectSettingsKycToJSON;
exports.ProjectSettingsPrivacyFromJSON = ProjectSettingsPrivacy.ProjectSettingsPrivacyFromJSON;
exports.ProjectSettingsPrivacyFromJSONTyped = ProjectSettingsPrivacy.ProjectSettingsPrivacyFromJSONTyped;
exports.ProjectSettingsPrivacyToJSON = ProjectSettingsPrivacy.ProjectSettingsPrivacyToJSON;
exports.ProjectSettingsSdkFromJSON = ProjectSettingsSdk.ProjectSettingsSdkFromJSON;
exports.ProjectSettingsSdkFromJSONTyped = ProjectSettingsSdk.ProjectSettingsSdkFromJSONTyped;
exports.ProjectSettingsSdkToJSON = ProjectSettingsSdk.ProjectSettingsSdkToJSON;
exports.ProjectSettingsSdkAccountAbstractionFromJSON = ProjectSettingsSdkAccountAbstraction.ProjectSettingsSdkAccountAbstractionFromJSON;
exports.ProjectSettingsSdkAccountAbstractionFromJSONTyped = ProjectSettingsSdkAccountAbstraction.ProjectSettingsSdkAccountAbstractionFromJSONTyped;
exports.ProjectSettingsSdkAccountAbstractionToJSON = ProjectSettingsSdkAccountAbstraction.ProjectSettingsSdkAccountAbstractionToJSON;
exports.ProjectSettingsSdkEmailSignInFromJSON = ProjectSettingsSdkEmailSignIn.ProjectSettingsSdkEmailSignInFromJSON;
exports.ProjectSettingsSdkEmailSignInFromJSONTyped = ProjectSettingsSdkEmailSignIn.ProjectSettingsSdkEmailSignInFromJSONTyped;
exports.ProjectSettingsSdkEmailSignInToJSON = ProjectSettingsSdkEmailSignIn.ProjectSettingsSdkEmailSignInToJSON;
exports.ProjectSettingsSdkEmbeddedWalletsFromJSON = ProjectSettingsSdkEmbeddedWallets.ProjectSettingsSdkEmbeddedWalletsFromJSON;
exports.ProjectSettingsSdkEmbeddedWalletsFromJSONTyped = ProjectSettingsSdkEmbeddedWallets.ProjectSettingsSdkEmbeddedWalletsFromJSONTyped;
exports.ProjectSettingsSdkEmbeddedWalletsToJSON = ProjectSettingsSdkEmbeddedWallets.ProjectSettingsSdkEmbeddedWalletsToJSON;
exports.ProjectSettingsSdkSocialSignInFromJSON = ProjectSettingsSdkSocialSignIn.ProjectSettingsSdkSocialSignInFromJSON;
exports.ProjectSettingsSdkSocialSignInFromJSONTyped = ProjectSettingsSdkSocialSignIn.ProjectSettingsSdkSocialSignInFromJSONTyped;
exports.ProjectSettingsSdkSocialSignInToJSON = ProjectSettingsSdkSocialSignIn.ProjectSettingsSdkSocialSignInToJSON;
exports.ProjectSettingsSdkWalletConnectFromJSON = ProjectSettingsSdkWalletConnect.ProjectSettingsSdkWalletConnectFromJSON;
exports.ProjectSettingsSdkWalletConnectFromJSONTyped = ProjectSettingsSdkWalletConnect.ProjectSettingsSdkWalletConnectFromJSONTyped;
exports.ProjectSettingsSdkWalletConnectToJSON = ProjectSettingsSdkWalletConnect.ProjectSettingsSdkWalletConnectToJSON;
exports.ProjectSettingsSecurityFromJSON = ProjectSettingsSecurity.ProjectSettingsSecurityFromJSON;
exports.ProjectSettingsSecurityFromJSONTyped = ProjectSettingsSecurity.ProjectSettingsSecurityFromJSONTyped;
exports.ProjectSettingsSecurityToJSON = ProjectSettingsSecurity.ProjectSettingsSecurityToJSON;
Object.defineProperty(exports, 'ProviderEntryPointVersionEnum', {
	enumerable: true,
	get: function () { return Provider.ProviderEntryPointVersionEnum; }
});
exports.ProviderFromJSON = Provider.ProviderFromJSON;
exports.ProviderFromJSONTyped = Provider.ProviderFromJSONTyped;
Object.defineProperty(exports, 'ProviderKernelVersionEnum', {
	enumerable: true,
	get: function () { return Provider.ProviderKernelVersionEnum; }
});
exports.ProviderToJSON = Provider.ProviderToJSON;
exports.ProviderAgreementFromJSON = ProviderAgreement.ProviderAgreementFromJSON;
exports.ProviderAgreementFromJSONTyped = ProviderAgreement.ProviderAgreementFromJSONTyped;
exports.ProviderAgreementToJSON = ProviderAgreement.ProviderAgreementToJSON;
Object.defineProperty(exports, 'ProviderEnum', {
	enumerable: true,
	get: function () { return ProviderEnum.ProviderEnum; }
});
exports.ProviderEnumFromJSON = ProviderEnum.ProviderEnumFromJSON;
exports.ProviderEnumFromJSONTyped = ProviderEnum.ProviderEnumFromJSONTyped;
exports.ProviderEnumToJSON = ProviderEnum.ProviderEnumToJSON;
exports.PublicKeyCredentialDescriptorFromJSON = PublicKeyCredentialDescriptor.PublicKeyCredentialDescriptorFromJSON;
exports.PublicKeyCredentialDescriptorFromJSONTyped = PublicKeyCredentialDescriptor.PublicKeyCredentialDescriptorFromJSONTyped;
exports.PublicKeyCredentialDescriptorToJSON = PublicKeyCredentialDescriptor.PublicKeyCredentialDescriptorToJSON;
exports.PublicKeyCredentialParametersFromJSON = PublicKeyCredentialParameters.PublicKeyCredentialParametersFromJSON;
exports.PublicKeyCredentialParametersFromJSONTyped = PublicKeyCredentialParameters.PublicKeyCredentialParametersFromJSONTyped;
exports.PublicKeyCredentialParametersToJSON = PublicKeyCredentialParameters.PublicKeyCredentialParametersToJSON;
exports.PublicKeyCredentialRpEntityFromJSON = PublicKeyCredentialRpEntity.PublicKeyCredentialRpEntityFromJSON;
exports.PublicKeyCredentialRpEntityFromJSONTyped = PublicKeyCredentialRpEntity.PublicKeyCredentialRpEntityFromJSONTyped;
exports.PublicKeyCredentialRpEntityToJSON = PublicKeyCredentialRpEntity.PublicKeyCredentialRpEntityToJSON;
Object.defineProperty(exports, 'PublicKeyCredentialType', {
	enumerable: true,
	get: function () { return PublicKeyCredentialType.PublicKeyCredentialType; }
});
exports.PublicKeyCredentialTypeFromJSON = PublicKeyCredentialType.PublicKeyCredentialTypeFromJSON;
exports.PublicKeyCredentialTypeFromJSONTyped = PublicKeyCredentialType.PublicKeyCredentialTypeFromJSONTyped;
exports.PublicKeyCredentialTypeToJSON = PublicKeyCredentialType.PublicKeyCredentialTypeToJSON;
exports.PublicKeyCredentialUserEntityFromJSON = PublicKeyCredentialUserEntity.PublicKeyCredentialUserEntityFromJSON;
exports.PublicKeyCredentialUserEntityFromJSONTyped = PublicKeyCredentialUserEntity.PublicKeyCredentialUserEntityFromJSONTyped;
exports.PublicKeyCredentialUserEntityToJSON = PublicKeyCredentialUserEntity.PublicKeyCredentialUserEntityToJSON;
exports.PublishEventsFromJSON = PublishEvents.PublishEventsFromJSON;
exports.PublishEventsFromJSONTyped = PublishEvents.PublishEventsFromJSONTyped;
exports.PublishEventsToJSON = PublishEvents.PublishEventsToJSON;
exports.PublishEventsEventsFromJSON = PublishEventsEvents.PublishEventsEventsFromJSON;
exports.PublishEventsEventsFromJSONTyped = PublishEventsEvents.PublishEventsEventsFromJSONTyped;
exports.PublishEventsEventsToJSON = PublishEventsEvents.PublishEventsEventsToJSON;
Object.defineProperty(exports, 'PublishEventsEventsTypeEnum', {
	enumerable: true,
	get: function () { return PublishEventsEvents.PublishEventsEventsTypeEnum; }
});
exports.ReactSettingsFromJSON = ReactSettings.ReactSettingsFromJSON;
exports.ReactSettingsFromJSONTyped = ReactSettings.ReactSettingsFromJSONTyped;
exports.ReactSettingsToJSON = ReactSettings.ReactSettingsToJSON;
exports.RegisterEmbeddedWalletSessionKeyResponseFromJSON = RegisterEmbeddedWalletSessionKeyResponse.RegisterEmbeddedWalletSessionKeyResponseFromJSON;
exports.RegisterEmbeddedWalletSessionKeyResponseFromJSONTyped = RegisterEmbeddedWalletSessionKeyResponse.RegisterEmbeddedWalletSessionKeyResponseFromJSONTyped;
exports.RegisterEmbeddedWalletSessionKeyResponseToJSON = RegisterEmbeddedWalletSessionKeyResponse.RegisterEmbeddedWalletSessionKeyResponseToJSON;
exports.RegisterSessionKeyRequestFromJSON = RegisterSessionKeyRequest.RegisterSessionKeyRequestFromJSON;
exports.RegisterSessionKeyRequestFromJSONTyped = RegisterSessionKeyRequest.RegisterSessionKeyRequestFromJSONTyped;
exports.RegisterSessionKeyRequestToJSON = RegisterSessionKeyRequest.RegisterSessionKeyRequestToJSON;
Object.defineProperty(exports, 'ResidentKeyRequirement', {
	enumerable: true,
	get: function () { return ResidentKeyRequirement.ResidentKeyRequirement; }
});
exports.ResidentKeyRequirementFromJSON = ResidentKeyRequirement.ResidentKeyRequirementFromJSON;
exports.ResidentKeyRequirementFromJSONTyped = ResidentKeyRequirement.ResidentKeyRequirementFromJSONTyped;
exports.ResidentKeyRequirementToJSON = ResidentKeyRequirement.ResidentKeyRequirementToJSON;
exports.ScanWebsiteUrlRequestFromJSON = ScanWebsiteUrlRequest.ScanWebsiteUrlRequestFromJSON;
exports.ScanWebsiteUrlRequestFromJSONTyped = ScanWebsiteUrlRequest.ScanWebsiteUrlRequestFromJSONTyped;
exports.ScanWebsiteUrlRequestToJSON = ScanWebsiteUrlRequest.ScanWebsiteUrlRequestToJSON;
exports.ScanWebsiteUrlResponseFromJSON = ScanWebsiteUrlResponse.ScanWebsiteUrlResponseFromJSON;
exports.ScanWebsiteUrlResponseFromJSONTyped = ScanWebsiteUrlResponse.ScanWebsiteUrlResponseFromJSONTyped;
exports.ScanWebsiteUrlResponseToJSON = ScanWebsiteUrlResponse.ScanWebsiteUrlResponseToJSON;
exports.SdkSettingsRequestFromJSON = SdkSettingsRequest.SdkSettingsRequestFromJSON;
exports.SdkSettingsRequestFromJSONTyped = SdkSettingsRequest.SdkSettingsRequestFromJSONTyped;
exports.SdkSettingsRequestToJSON = SdkSettingsRequest.SdkSettingsRequestToJSON;
exports.SdkUserFromJSON = SdkUser.SdkUserFromJSON;
exports.SdkUserFromJSONTyped = SdkUser.SdkUserFromJSONTyped;
exports.SdkUserToJSON = SdkUser.SdkUserToJSON;
exports.SdkViewFromJSON = SdkView.SdkViewFromJSON;
exports.SdkViewFromJSONTyped = SdkView.SdkViewFromJSONTyped;
exports.SdkViewToJSON = SdkView.SdkViewToJSON;
exports.SdkViewSectionFromJSON = SdkViewSection.SdkViewSectionFromJSON;
exports.SdkViewSectionFromJSONTyped = SdkViewSection.SdkViewSectionFromJSONTyped;
exports.SdkViewSectionToJSON = SdkViewSection.SdkViewSectionToJSON;
Object.defineProperty(exports, 'SdkViewSectionAlignment', {
	enumerable: true,
	get: function () { return SdkViewSectionAlignment.SdkViewSectionAlignment; }
});
exports.SdkViewSectionAlignmentFromJSON = SdkViewSectionAlignment.SdkViewSectionAlignmentFromJSON;
exports.SdkViewSectionAlignmentFromJSONTyped = SdkViewSectionAlignment.SdkViewSectionAlignmentFromJSONTyped;
exports.SdkViewSectionAlignmentToJSON = SdkViewSectionAlignment.SdkViewSectionAlignmentToJSON;
Object.defineProperty(exports, 'SdkViewSectionType', {
	enumerable: true,
	get: function () { return SdkViewSectionType.SdkViewSectionType; }
});
exports.SdkViewSectionTypeFromJSON = SdkViewSectionType.SdkViewSectionTypeFromJSON;
exports.SdkViewSectionTypeFromJSONTyped = SdkViewSectionType.SdkViewSectionTypeFromJSONTyped;
exports.SdkViewSectionTypeToJSON = SdkViewSectionType.SdkViewSectionTypeToJSON;
Object.defineProperty(exports, 'SdkViewType', {
	enumerable: true,
	get: function () { return SdkViewType.SdkViewType; }
});
exports.SdkViewTypeFromJSON = SdkViewType.SdkViewTypeFromJSON;
exports.SdkViewTypeFromJSONTyped = SdkViewType.SdkViewTypeFromJSONTyped;
exports.SdkViewTypeToJSON = SdkViewType.SdkViewTypeToJSON;
exports.SdkViewUpdateRequestFromJSON = SdkViewUpdateRequest.SdkViewUpdateRequestFromJSON;
exports.SdkViewUpdateRequestFromJSONTyped = SdkViewUpdateRequest.SdkViewUpdateRequestFromJSONTyped;
exports.SdkViewUpdateRequestToJSON = SdkViewUpdateRequest.SdkViewUpdateRequestToJSON;
exports.SdkViewsResponseFromJSON = SdkViewsResponse.SdkViewsResponseFromJSON;
exports.SdkViewsResponseFromJSONTyped = SdkViewsResponse.SdkViewsResponseFromJSONTyped;
exports.SdkViewsResponseToJSON = SdkViewsResponse.SdkViewsResponseToJSON;
Object.defineProperty(exports, 'SignInProviderEnum', {
	enumerable: true,
	get: function () { return SignInProviderEnum.SignInProviderEnum; }
});
exports.SignInProviderEnumFromJSON = SignInProviderEnum.SignInProviderEnumFromJSON;
exports.SignInProviderEnumFromJSONTyped = SignInProviderEnum.SignInProviderEnumFromJSONTyped;
exports.SignInProviderEnumToJSON = SignInProviderEnum.SignInProviderEnumToJSON;
exports.SmsCountryCodeFromJSON = SmsCountryCode.SmsCountryCodeFromJSON;
exports.SmsCountryCodeFromJSONTyped = SmsCountryCode.SmsCountryCodeFromJSONTyped;
exports.SmsCountryCodeToJSON = SmsCountryCode.SmsCountryCodeToJSON;
exports.SmsVerificationCreateRequestFromJSON = SmsVerificationCreateRequest.SmsVerificationCreateRequestFromJSON;
exports.SmsVerificationCreateRequestFromJSONTyped = SmsVerificationCreateRequest.SmsVerificationCreateRequestFromJSONTyped;
exports.SmsVerificationCreateRequestToJSON = SmsVerificationCreateRequest.SmsVerificationCreateRequestToJSON;
exports.SmsVerificationCreateResponseFromJSON = SmsVerificationCreateResponse.SmsVerificationCreateResponseFromJSON;
exports.SmsVerificationCreateResponseFromJSONTyped = SmsVerificationCreateResponse.SmsVerificationCreateResponseFromJSONTyped;
exports.SmsVerificationCreateResponseToJSON = SmsVerificationCreateResponse.SmsVerificationCreateResponseToJSON;
exports.SmsVerificationRetryRequestFromJSON = SmsVerificationRetryRequest.SmsVerificationRetryRequestFromJSON;
exports.SmsVerificationRetryRequestFromJSONTyped = SmsVerificationRetryRequest.SmsVerificationRetryRequestFromJSONTyped;
exports.SmsVerificationRetryRequestToJSON = SmsVerificationRetryRequest.SmsVerificationRetryRequestToJSON;
exports.SmsVerificationVerifyRequestFromJSON = SmsVerificationVerifyRequest.SmsVerificationVerifyRequestFromJSON;
exports.SmsVerificationVerifyRequestFromJSONTyped = SmsVerificationVerifyRequest.SmsVerificationVerifyRequestFromJSONTyped;
exports.SmsVerificationVerifyRequestToJSON = SmsVerificationVerifyRequest.SmsVerificationVerifyRequestToJSON;
exports.SocialSignInProviderFromJSON = SocialSignInProvider.SocialSignInProviderFromJSON;
exports.SocialSignInProviderFromJSONTyped = SocialSignInProvider.SocialSignInProviderFromJSONTyped;
exports.SocialSignInProviderToJSON = SocialSignInProvider.SocialSignInProviderToJSON;
Object.defineProperty(exports, 'SocialSignInProviderEnum', {
	enumerable: true,
	get: function () { return SocialSignInProviderEnum.SocialSignInProviderEnum; }
});
exports.SocialSignInProviderEnumFromJSON = SocialSignInProviderEnum.SocialSignInProviderEnumFromJSON;
exports.SocialSignInProviderEnumFromJSONTyped = SocialSignInProviderEnum.SocialSignInProviderEnumFromJSONTyped;
exports.SocialSignInProviderEnumToJSON = SocialSignInProviderEnum.SocialSignInProviderEnumToJSON;
exports.SolanaTransactionOptimizationRequestFromJSON = SolanaTransactionOptimizationRequest.SolanaTransactionOptimizationRequestFromJSON;
exports.SolanaTransactionOptimizationRequestFromJSONTyped = SolanaTransactionOptimizationRequest.SolanaTransactionOptimizationRequestFromJSONTyped;
exports.SolanaTransactionOptimizationRequestToJSON = SolanaTransactionOptimizationRequest.SolanaTransactionOptimizationRequestToJSON;
exports.SolanaTransactionOptimizationResponseFromJSON = SolanaTransactionOptimizationResponse.SolanaTransactionOptimizationResponseFromJSON;
exports.SolanaTransactionOptimizationResponseFromJSONTyped = SolanaTransactionOptimizationResponse.SolanaTransactionOptimizationResponseFromJSONTyped;
exports.SolanaTransactionOptimizationResponseToJSON = SolanaTransactionOptimizationResponse.SolanaTransactionOptimizationResponseToJSON;
exports.SupportedOnrampsResponseFromJSON = SupportedOnrampsResponse.SupportedOnrampsResponseFromJSON;
exports.SupportedOnrampsResponseFromJSONTyped = SupportedOnrampsResponse.SupportedOnrampsResponseFromJSONTyped;
exports.SupportedOnrampsResponseToJSON = SupportedOnrampsResponse.SupportedOnrampsResponseToJSON;
exports.SupportedSecurityMethodFromJSON = SupportedSecurityMethod.SupportedSecurityMethodFromJSON;
exports.SupportedSecurityMethodFromJSONTyped = SupportedSecurityMethod.SupportedSecurityMethodFromJSONTyped;
exports.SupportedSecurityMethodToJSON = SupportedSecurityMethod.SupportedSecurityMethodToJSON;
exports.SupportedSecurityMethodsFromJSON = SupportedSecurityMethods.SupportedSecurityMethodsFromJSON;
exports.SupportedSecurityMethodsFromJSONTyped = SupportedSecurityMethods.SupportedSecurityMethodsFromJSONTyped;
exports.SupportedSecurityMethodsToJSON = SupportedSecurityMethods.SupportedSecurityMethodsToJSON;
exports.TelegramPostRequestFromJSON = TelegramPostRequest.TelegramPostRequestFromJSON;
exports.TelegramPostRequestFromJSONTyped = TelegramPostRequest.TelegramPostRequestFromJSONTyped;
exports.TelegramPostRequestToJSON = TelegramPostRequest.TelegramPostRequestToJSON;
exports.TelegramUserFromJSON = TelegramUser.TelegramUserFromJSON;
exports.TelegramUserFromJSONTyped = TelegramUser.TelegramUserFromJSONTyped;
exports.TelegramUserToJSON = TelegramUser.TelegramUserToJSON;
Object.defineProperty(exports, 'TimeUnitEnum', {
	enumerable: true,
	get: function () { return TimeUnitEnum.TimeUnitEnum; }
});
exports.TimeUnitEnumFromJSON = TimeUnitEnum.TimeUnitEnumFromJSON;
exports.TimeUnitEnumFromJSONTyped = TimeUnitEnum.TimeUnitEnumFromJSONTyped;
exports.TimeUnitEnumToJSON = TimeUnitEnum.TimeUnitEnumToJSON;
exports.TokenBalanceFromJSON = TokenBalance.TokenBalanceFromJSON;
exports.TokenBalanceFromJSONTyped = TokenBalance.TokenBalanceFromJSONTyped;
exports.TokenBalanceToJSON = TokenBalance.TokenBalanceToJSON;
exports.TooManyRequestsFromJSON = TooManyRequests.TooManyRequestsFromJSON;
exports.TooManyRequestsFromJSONTyped = TooManyRequests.TooManyRequestsFromJSONTyped;
exports.TooManyRequestsToJSON = TooManyRequests.TooManyRequestsToJSON;
exports.TurnkeyCreateWalletAccountsRequestBodyFromJSON = TurnkeyCreateWalletAccountsRequestBody.TurnkeyCreateWalletAccountsRequestBodyFromJSON;
exports.TurnkeyCreateWalletAccountsRequestBodyFromJSONTyped = TurnkeyCreateWalletAccountsRequestBody.TurnkeyCreateWalletAccountsRequestBodyFromJSONTyped;
exports.TurnkeyCreateWalletAccountsRequestBodyToJSON = TurnkeyCreateWalletAccountsRequestBody.TurnkeyCreateWalletAccountsRequestBodyToJSON;
exports.TurnkeyCreateWalletAccountsRequestBodyParametersFromJSON = TurnkeyCreateWalletAccountsRequestBodyParameters.TurnkeyCreateWalletAccountsRequestBodyParametersFromJSON;
exports.TurnkeyCreateWalletAccountsRequestBodyParametersFromJSONTyped = TurnkeyCreateWalletAccountsRequestBodyParameters.TurnkeyCreateWalletAccountsRequestBodyParametersFromJSONTyped;
exports.TurnkeyCreateWalletAccountsRequestBodyParametersToJSON = TurnkeyCreateWalletAccountsRequestBodyParameters.TurnkeyCreateWalletAccountsRequestBodyParametersToJSON;
exports.TurnkeySignedRequestFromJSON = TurnkeySignedRequest.TurnkeySignedRequestFromJSON;
exports.TurnkeySignedRequestFromJSONTyped = TurnkeySignedRequest.TurnkeySignedRequestFromJSONTyped;
exports.TurnkeySignedRequestToJSON = TurnkeySignedRequest.TurnkeySignedRequestToJSON;
exports.TurnkeyStampFromJSON = TurnkeyStamp.TurnkeyStampFromJSON;
exports.TurnkeyStampFromJSONTyped = TurnkeyStamp.TurnkeyStampFromJSONTyped;
exports.TurnkeyStampToJSON = TurnkeyStamp.TurnkeyStampToJSON;
exports.TurnkeyWalletAccountFromJSON = TurnkeyWalletAccount.TurnkeyWalletAccountFromJSON;
exports.TurnkeyWalletAccountFromJSONTyped = TurnkeyWalletAccount.TurnkeyWalletAccountFromJSONTyped;
exports.TurnkeyWalletAccountToJSON = TurnkeyWalletAccount.TurnkeyWalletAccountToJSON;
exports.TurnkeyWalletPropertiesFromJSON = TurnkeyWalletProperties.TurnkeyWalletPropertiesFromJSON;
exports.TurnkeyWalletPropertiesFromJSONTyped = TurnkeyWalletProperties.TurnkeyWalletPropertiesFromJSONTyped;
exports.TurnkeyWalletPropertiesToJSON = TurnkeyWalletProperties.TurnkeyWalletPropertiesToJSON;
exports.UnauthorizedFromJSON = Unauthorized.UnauthorizedFromJSON;
exports.UnauthorizedFromJSONTyped = Unauthorized.UnauthorizedFromJSONTyped;
exports.UnauthorizedToJSON = Unauthorized.UnauthorizedToJSON;
exports.UnprocessableEntityFromJSON = UnprocessableEntity.UnprocessableEntityFromJSON;
exports.UnprocessableEntityFromJSONTyped = UnprocessableEntity.UnprocessableEntityFromJSONTyped;
exports.UnprocessableEntityToJSON = UnprocessableEntity.UnprocessableEntityToJSON;
Object.defineProperty(exports, 'UnprocessableEntityErrorCode', {
	enumerable: true,
	get: function () { return UnprocessableEntityErrorCode.UnprocessableEntityErrorCode; }
});
exports.UnprocessableEntityErrorCodeFromJSON = UnprocessableEntityErrorCode.UnprocessableEntityErrorCodeFromJSON;
exports.UnprocessableEntityErrorCodeFromJSONTyped = UnprocessableEntityErrorCode.UnprocessableEntityErrorCodeFromJSONTyped;
exports.UnprocessableEntityErrorCodeToJSON = UnprocessableEntityErrorCode.UnprocessableEntityErrorCodeToJSON;
exports.UnprocessableEntityErrorPayloadFromJSON = UnprocessableEntityErrorPayload.UnprocessableEntityErrorPayloadFromJSON;
exports.UnprocessableEntityErrorPayloadFromJSONTyped = UnprocessableEntityErrorPayload.UnprocessableEntityErrorPayloadFromJSONTyped;
exports.UnprocessableEntityErrorPayloadToJSON = UnprocessableEntityErrorPayload.UnprocessableEntityErrorPayloadToJSON;
exports.UpdateRecoveryEmailRequestFromJSON = UpdateRecoveryEmailRequest.UpdateRecoveryEmailRequestFromJSON;
exports.UpdateRecoveryEmailRequestFromJSONTyped = UpdateRecoveryEmailRequest.UpdateRecoveryEmailRequestFromJSONTyped;
exports.UpdateRecoveryEmailRequestToJSON = UpdateRecoveryEmailRequest.UpdateRecoveryEmailRequestToJSON;
exports.UpdateSelfResponseFromJSON = UpdateSelfResponse.UpdateSelfResponseFromJSON;
exports.UpdateSelfResponseFromJSONTyped = UpdateSelfResponse.UpdateSelfResponseFromJSONTyped;
exports.UpdateSelfResponseToJSON = UpdateSelfResponse.UpdateSelfResponseToJSON;
exports.UpdateSelfResponseAllOfFromJSON = UpdateSelfResponseAllOf.UpdateSelfResponseAllOfFromJSON;
exports.UpdateSelfResponseAllOfFromJSONTyped = UpdateSelfResponseAllOf.UpdateSelfResponseAllOfFromJSONTyped;
exports.UpdateSelfResponseAllOfToJSON = UpdateSelfResponseAllOf.UpdateSelfResponseAllOfToJSON;
exports.UpdateUserPasskeyRequestFromJSON = UpdateUserPasskeyRequest.UpdateUserPasskeyRequestFromJSON;
exports.UpdateUserPasskeyRequestFromJSONTyped = UpdateUserPasskeyRequest.UpdateUserPasskeyRequestFromJSONTyped;
exports.UpdateUserPasskeyRequestToJSON = UpdateUserPasskeyRequest.UpdateUserPasskeyRequestToJSON;
exports.UserFieldsFromJSON = UserFields.UserFieldsFromJSON;
exports.UserFieldsFromJSONTyped = UserFields.UserFieldsFromJSONTyped;
exports.UserFieldsToJSON = UserFields.UserFieldsToJSON;
Object.defineProperty(exports, 'UserIdentifierTypeEnum', {
	enumerable: true,
	get: function () { return UserIdentifierTypeEnum.UserIdentifierTypeEnum; }
});
exports.UserIdentifierTypeEnumFromJSON = UserIdentifierTypeEnum.UserIdentifierTypeEnumFromJSON;
exports.UserIdentifierTypeEnumFromJSONTyped = UserIdentifierTypeEnum.UserIdentifierTypeEnumFromJSONTyped;
exports.UserIdentifierTypeEnumToJSON = UserIdentifierTypeEnum.UserIdentifierTypeEnumToJSON;
exports.UserOauthAccessTokenResponseFromJSON = UserOauthAccessTokenResponse.UserOauthAccessTokenResponseFromJSON;
exports.UserOauthAccessTokenResponseFromJSONTyped = UserOauthAccessTokenResponse.UserOauthAccessTokenResponseFromJSONTyped;
exports.UserOauthAccessTokenResponseToJSON = UserOauthAccessTokenResponse.UserOauthAccessTokenResponseToJSON;
exports.UserPasskeyFromJSON = UserPasskey.UserPasskeyFromJSON;
exports.UserPasskeyFromJSONTyped = UserPasskey.UserPasskeyFromJSONTyped;
exports.UserPasskeyToJSON = UserPasskey.UserPasskeyToJSON;
Object.defineProperty(exports, 'UserVerificationRequirement', {
	enumerable: true,
	get: function () { return UserVerificationRequirement.UserVerificationRequirement; }
});
exports.UserVerificationRequirementFromJSON = UserVerificationRequirement.UserVerificationRequirementFromJSON;
exports.UserVerificationRequirementFromJSONTyped = UserVerificationRequirement.UserVerificationRequirementFromJSONTyped;
exports.UserVerificationRequirementToJSON = UserVerificationRequirement.UserVerificationRequirementToJSON;
exports.UserWalletSelectionRequestFromJSON = UserWalletSelectionRequest.UserWalletSelectionRequestFromJSON;
exports.UserWalletSelectionRequestFromJSONTyped = UserWalletSelectionRequest.UserWalletSelectionRequestFromJSONTyped;
exports.UserWalletSelectionRequestToJSON = UserWalletSelectionRequest.UserWalletSelectionRequestToJSON;
exports.VerifyRequestFromJSON = VerifyRequest.VerifyRequestFromJSON;
exports.VerifyRequestFromJSONTyped = VerifyRequest.VerifyRequestFromJSONTyped;
exports.VerifyRequestToJSON = VerifyRequest.VerifyRequestToJSON;
exports.VerifyResponseFromJSON = VerifyResponse.VerifyResponseFromJSON;
exports.VerifyResponseFromJSONTyped = VerifyResponse.VerifyResponseFromJSONTyped;
exports.VerifyResponseToJSON = VerifyResponse.VerifyResponseToJSON;
exports.VerifyUnlinkRequestFromJSON = VerifyUnlinkRequest.VerifyUnlinkRequestFromJSON;
exports.VerifyUnlinkRequestFromJSONTyped = VerifyUnlinkRequest.VerifyUnlinkRequestFromJSONTyped;
exports.VerifyUnlinkRequestToJSON = VerifyUnlinkRequest.VerifyUnlinkRequestToJSON;
exports.WalletAdditionalAddressFromJSON = WalletAdditionalAddress.WalletAdditionalAddressFromJSON;
exports.WalletAdditionalAddressFromJSONTyped = WalletAdditionalAddress.WalletAdditionalAddressFromJSONTyped;
exports.WalletAdditionalAddressToJSON = WalletAdditionalAddress.WalletAdditionalAddressToJSON;
Object.defineProperty(exports, 'WalletAddressType', {
	enumerable: true,
	get: function () { return WalletAddressType.WalletAddressType; }
});
exports.WalletAddressTypeFromJSON = WalletAddressType.WalletAddressTypeFromJSON;
exports.WalletAddressTypeFromJSONTyped = WalletAddressType.WalletAddressTypeFromJSONTyped;
exports.WalletAddressTypeToJSON = WalletAddressType.WalletAddressTypeToJSON;
exports.WalletPropertiesFromJSON = WalletProperties.WalletPropertiesFromJSON;
exports.WalletPropertiesFromJSONTyped = WalletProperties.WalletPropertiesFromJSONTyped;
exports.WalletPropertiesToJSON = WalletProperties.WalletPropertiesToJSON;
Object.defineProperty(exports, 'WalletProviderEnum', {
	enumerable: true,
	get: function () { return WalletProviderEnum.WalletProviderEnum; }
});
exports.WalletProviderEnumFromJSON = WalletProviderEnum.WalletProviderEnumFromJSON;
exports.WalletProviderEnumFromJSONTyped = WalletProviderEnum.WalletProviderEnumFromJSONTyped;
exports.WalletProviderEnumToJSON = WalletProviderEnum.WalletProviderEnumToJSON;
