// @turnkey/sdk-server/dist/index.js
'use strict';

var apiKeyStamper = require('@turnkey/api-key-stamper');
var http = require('@turnkey/http');
var sdkClient = require('./sdk-client.js');
var turnkeyHelpers = require('./turnkey-helpers.js');
var universal = require('./universal.js');



Object.defineProperty(exports, 'ApiKeyStamper', {
	enumerable: true,
	get: function () { return apiKeyStamper.ApiKeyStamper; }
});
Object.defineProperty(exports, 'signWithApiKey', {
	enumerable: true,
	get: function () { return apiKeyStamper.signWithApiKey; }
});
Object.defineProperty(exports, 'TurnkeyActivityError', {
	enumerable: true,
	get: function () { return http.TurnkeyActivityError; }
});
Object.defineProperty(exports, 'TurnkeyApi', {
	enumerable: true,
	get: function () { return http.TurnkeyApi; }
});
Object.defineProperty(exports, 'TurnkeyRequestError', {
	enumerable: true,
	get: function () { return http.TurnkeyRequestError; }
});
Object.defineProperty(exports, 'createActivityPoller', {
	enumerable: true,
	get: function () { return http.createActivityPoller; }
});
Object.defineProperty(exports, 'getWebAuthnAttestation', {
	enumerable: true,
	get: function () { return http.getWebAuthnAttestation; }
});
Object.defineProperty(exports, 'sealAndStampRequestBody', {
	enumerable: true,
	get: function () { return http.sealAndStampRequestBody; }
});
exports.Turnkey = sdkClient.TurnkeyServerSDK;
exports.TurnkeyApiClient = sdkClient.TurnkeyApiClient;
exports.TurnkeyServerClient = sdkClient.TurnkeyServerClient;
exports.DEFAULT_ETHEREUM_ACCOUNTS = turnkeyHelpers.DEFAULT_ETHEREUM_ACCOUNTS;
exports.defaultEthereumAccountAtIndex = turnkeyHelpers.defaultEthereumAccountAtIndex;
exports.fetch = universal.fetch;
