// @turnkey/sdk-browser/dist/index.js
'use strict';

var http = require('@turnkey/http');
var apiKeyStamper = require('@turnkey/api-key-stamper');
var iframeStamper = require('@turnkey/iframe-stamper');
var webauthnStamper = require('@turnkey/webauthn-stamper');
var sdkClient = require('./sdk-client.js');
var turnkeyHelpers = require('./turnkey-helpers.js');



Object.defineProperty(exports, 'TurnkeyActivityError', {
	enumerable: true,
	get: function () { return http.TurnkeyActivityError; }
});
Object.defineProperty(exports, 'TurnkeyApi', {
	enumerable: true,
	get: function () { return http.TurnkeyApi; }
});
Object.defineProperty(exports, 'TurnkeyRequestError', {
	enumerable: true,
	get: function () { return http.TurnkeyRequestError; }
});
Object.defineProperty(exports, 'createActivityPoller', {
	enumerable: true,
	get: function () { return http.createActivityPoller; }
});
Object.defineProperty(exports, 'getWebAuthnAttestation', {
	enumerable: true,
	get: function () { return http.getWebAuthnAttestation; }
});
Object.defineProperty(exports, 'sealAndStampRequestBody', {
	enumerable: true,
	get: function () { return http.sealAndStampRequestBody; }
});
Object.defineProperty(exports, 'ApiKeyStamper', {
	enumerable: true,
	get: function () { return apiKeyStamper.ApiKeyStamper; }
});
Object.defineProperty(exports, 'signWithApiKey', {
	enumerable: true,
	get: function () { return apiKeyStamper.signWithApiKey; }
});
Object.defineProperty(exports, 'IframeEventType', {
	enumerable: true,
	get: function () { return iframeStamper.IframeEventType; }
});
Object.defineProperty(exports, 'IframeStamper', {
	enumerable: true,
	get: function () { return iframeStamper.IframeStamper; }
});
Object.defineProperty(exports, 'WebauthnStamper', {
	enumerable: true,
	get: function () { return webauthnStamper.WebauthnStamper; }
});
exports.Turnkey = sdkClient.TurnkeyBrowserSDK;
exports.TurnkeyBrowserClient = sdkClient.TurnkeyBrowserClient;
exports.TurnkeyIframeClient = sdkClient.TurnkeyIframeClient;
exports.TurnkeyPasskeyClient = sdkClient.TurnkeyPasskeyClient;
exports.DEFAULT_ETHEREUM_ACCOUNTS = turnkeyHelpers.DEFAULT_ETHEREUM_ACCOUNTS;
exports.DEFAULT_SOLANA_ACCOUNTS = turnkeyHelpers.DEFAULT_SOLANA_ACCOUNTS;
exports.defaultEthereumAccountAtIndex = turnkeyHelpers.defaultEthereumAccountAtIndex;
exports.defaultSolanaAccountAtIndex = turnkeyHelpers.defaultSolanaAccountAtIndex;
